<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Morgante" />

<meta name="date" content="2025-04-16" />

<title>Genomic prediction of life span with different models</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp_lifespan_gxe</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/morgante_lab/dgrp_lifespan_gxe">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Genomic prediction of life span with
different models</h1>
<h4 class="author">Fabio Morgante</h4>
<h4 class="date">April 16, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp_lifespan_gxe/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241004code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241004code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241004)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetreee0a570b4e4a54d8ca4768bb69fd6027fb960b59ftargetblanke0a570ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/e0a570b4e4a54d8ca4768bb69fd6027fb960b59f" target="_blank">e0a570b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetreee0a570b4e4a54d8ca4768bb69fd6027fb960b59ftargetblanke0a570ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/e0a570b4e4a54d8ca4768bb69fd6027fb960b59f" target="_blank">e0a570b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .snakemake/
    Ignored:    output/E_fit/
    Ignored:    output/E_het_var_fit/
    Ignored:    output/G_fit/
    Ignored:    output/G_het_var_fit/
    Ignored:    output/GandE_fit/
    Ignored:    output/GandE_het_var_fit/
    Ignored:    output/GxE_by_chr_fit/
    Ignored:    output/GxE_by_chr_het_var_fit/
    Ignored:    output/GxE_fit/
    Ignored:    output/GxE_gwas/
    Ignored:    output/GxE_gwas_fit/
    Ignored:    output/GxE_gwas_het_var_fit/
    Ignored:    output/GxE_het_var_fit/
    Ignored:    output/mvgblup_fit/
    Ignored:    output/rrm_fit/
    Ignored:    output/test_sets/
    Ignored:    run/
    Ignored:    slurm/

Untracked files:
    Untracked:  analysis/dgrp_lifespan_gxe_genomic_results_OLD.Rmd
    Untracked:  code/fit_gwas_gxe_2envs_test.R
    Untracked:  code/fit_gwas_gxe_test.R
    Untracked:  code/fit_multitrait_gblup_whole_data.R
    Untracked:  scripts/run_fit_gwas_gxe_grammar_indicator.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_means.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_sex.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_temp.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lm.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_2envs_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_indicator.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_indicator_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_means.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_means_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_scaled_within_env_pheno_2envs_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_sex.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_sex_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_temp.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_temp_stand.sbatch

Unstaged changes:
    Modified:   dgrp_lifespan_gxe.yaml
    Modified:   dgrp_lifespan_gxe_snakefile
    Modified:   dgrp_lifespan_gxe_snakemake_submitter.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/dgrp_lifespan_gxe_genomic_results.Rmd</code>) and HTML
(<code>docs/dgrp_lifespan_gxe_genomic_results.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/e0a570b4e4a54d8ca4768bb69fd6027fb960b59f/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">e0a570b</a>
</td>
<td>
fmorgante
</td>
<td>
2025-04-16
</td>
<td>
Update results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/21039984de2ff72100b058b2f3280f79c499ac2e/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">2103998</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/18158a52319dfdf2f37803c356da3a20f7e3873e/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">18158a5</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
<td>
Minor fixes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/b69aff819f2f8163d4c66efc64f363d559139e9f/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">b69aff8</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/ac30118ace3ed69eb651512e27367fa05a52b544/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">ac30118</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
<td>
Minor fixes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/ec76fc861d9f9cd7de140202a1a7952e36a829f4/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">ec76fc8</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/583e8ce3141f44534bf030f98122066fd76a888b/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">583e8ce</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
<td>
Add RRM results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/9b44f8d85c07220ce84ab313a001ab16bd8757c9/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">9b44f8d</a>
</td>
<td>
fmorgante
</td>
<td>
2025-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/e4fa0f2bcb40cd22ea49909822db207dfc11a291/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">e4fa0f2</a>
</td>
<td>
fmorgante
</td>
<td>
2025-02-27
</td>
<td>
Add GxE-GWAS and by chr results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/86e70e8f90276892cf04e441b9537e217c723826/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/52b983f29198fd75a765214854dc956166c9fecc/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">52b983f</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
<td>
Add het var results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/39ba942039ab6b2be5a263b35b5c3601920665d1/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">39ba942</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
<td>
Add new results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/9a93c45b800f43a4f4c6d14619221bf65971f03e/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">9a93c45</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/cc02d0d573816c5d58468d6311a982274f3be934/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">cc02d0d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
<td>
Add results of new temp/sex scenario
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/b7f69008efb9bf3dff0435a11b27e2f5107ce07e/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">b7f6900</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/d4fe1263e221619e0dc9c12c7c9b6c2d2a7b5096/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">d4fe126</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Add descriptions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/a108f2043fb03f898a50d62f39fe6c00f21db686/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">a108f20</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Add results with different scaling of y
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/5247c8b6f6b9e21e549da947eb11e5513695abc2/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">5247c8b</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/66afcdd827caf63326608f0f2a09dc2f0ea3a2c3/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">66afcdd</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/f49ca0a50356e06fa5b2d8c0531eeef0980fe4ea/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">f49ca0a</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-24
</td>
<td>
Add across tempreature results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/bfc1461a735460ba936cfd01ccd96fccd70a08b3/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">bfc1461</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/db9c383061548e3043f9741fe822a0c3d6725f12/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">db9c383</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/1091563f08d32456638eee023d61a2ecf09576ef/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">1091563</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Minor fixes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/dce17d9d41f5075cccb08dd7584467fb186b8a85/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">dce17d9</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9558bc7af36bf7ad97e5faeeecc24a5ba7e87246/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">9558bc7</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Use R2 instead of r in genomic results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/8530bb379906a619955b9046e9d54820b473503a/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">8530bb3</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/7aecac219990c05ac11be6910c7e0f3a97f3667b/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">7aecac2</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Add PVE to genomic results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/77678919a17a358a8186f04be2ae3688ee0d1d70/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">7767891</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/24fb66c0b6810d35bbd8eb641186407bf7d2abd0/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">24fb66c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Update results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>###Load libraries
library(ggplot2)
library(cowplot)
library(tidyr)
library(dplyr)

prefix &lt;- &quot;dgrp_lifespan_gxe&quot;
reps &lt;- c(1,2,4,5,7,9)</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The goal of this analysis is to assess the performance of the linear
mixed models including genotype only, environment only, genotype +
environment, genotype + environment + genotype <span
class="math inline">\(\times\)</span> environment information at
predicting life span in the DGRP.</p>
<p>We used data from <a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000645">Huang
et al. (2020)</a>, which has life span measurements for <span
class="math inline">\(n=176\)</span> lines at 3 temperatures (18C, 25C,
28C) for the 2 sexes (coded as 1=females and 0=males), which results in
<span class="math inline">\(q=1,056\)</span> observations. The authors
showed that there is extensive <span class="math inline">\(G \times
E\)</span> affecting life span. In the present work, we used both
temperature and sex as <span class="math inline">\(c=2\)</span>
environmental variables, for a total of <span
class="math inline">\(r=6\)</span> environments.</p>
<p>We fitted the following linear mixed models:</p>
<ul>
<li>G-BLUP – <span class="math inline">\(y = \mu + g +
\epsilon\)</span>.</li>
<li>E-BLUP – <span class="math inline">\(y = \mu + e +
\epsilon\)</span>.</li>
<li>GE-BLUP – <span class="math inline">\(y = \mu + g + e +
\epsilon\)</span>.</li>
<li>GxE-BLUP – <span class="math inline">\(y = \mu + g + e + ge +
\epsilon\)</span>.</li>
</ul>
<p>where</p>
<p><span class="math inline">\(y\)</span> is a <span
class="math inline">\(q\)</span>-vector of phenotypic observations,
<span class="math inline">\(\mu\)</span> is a <span
class="math inline">\(q\)</span>-vector of intercept values, <span
class="math inline">\(g\)</span> is a <span
class="math inline">\(q\)</span>-vector of additive genetic values,
<span class="math inline">\(g \sim N_q(\mathbf O, \mathbf{ZGZ^\intercal}
\sigma^2_g)\)</span>, <span class="math inline">\(\mathbf Z\)</span> is
a <span class="math inline">\(q \times n\)</span> incidence matrix,
<span class="math inline">\(\mathbf G\)</span> is a <span
class="math inline">\(n \times n\)</span> genomic relationship matrix
(GRM), <span class="math inline">\(e\)</span> is a <span
class="math inline">\(q\)</span>-vector of environmental values, <span
class="math inline">\(e \sim N_q(\mathbf O, \mathbf{E}
\sigma^2_e)\)</span>, <span class="math inline">\(\mathbf E\)</span> is
a <span class="math inline">\(q \times q\)</span> matrix of similarity
based on environmental variables, computed as <span
class="math inline">\(\mathbf E = \mathbf{XX}^\intercal\)</span>, <span
class="math inline">\(\mathbf X\)</span> is a <span
class="math inline">\(q \times c\)</span> matrix of environmental
measurements, <span class="math inline">\(ge \sim N_q(\mathbf O,
\mathbf{ZGZ^\intercal \circ E} \sigma^2_{ge})\)</span>, <span
class="math inline">\(\epsilon\)</span> is a <span
class="math inline">\(q\)</span>-vector of residual values, <span
class="math inline">\(\epsilon \sim N_q(\mathbf O, \mathbf{I}
\sigma^2_\epsilon)\)</span>.</p>
<p>These models were also implemented with heterogeneous residual
variances across environments.</p>
<p>We also fit a multivariate GBLUP – <span class="math inline">\(Y = M
+ U + R\)</span>:</p>
<p>where</p>
<p><span class="math inline">\(Y\)</span> is an <span
class="math inline">\(n \times r\)</span> matrix of phenotypic
observations, <span class="math inline">\(M\)</span> is an <span
class="math inline">\(r\)</span>-vector of intercept values, <span
class="math inline">\(U\)</span> is an <span class="math inline">\(n
\times r\)</span> matrix of additive genetic values, <span
class="math inline">\(U \sim MN_{n \times r} (\mathbf O, \mathbf G,
\mathbf \Sigma_U)\)</span> and <span class="math inline">\(\mathbf
\Sigma_U\)</span> is an <span class="math inline">\(r \times r\)</span>
covariance matrix, <span class="math inline">\(R\)</span> is an <span
class="math inline">\(r \times r\)</span> matrix of residual values,
<span class="math inline">\(R \sim MN_{n \times r} (\mathbf O, \mathbf
I_n, \mathbf \Sigma_R)\)</span> and <span class="math inline">\(\mathbf
\Sigma_R\)</span> is an <span class="math inline">\(r \times r\)</span>
covariance matrix.</p>
<p>We also fit a random regression model – <span
class="math inline">\(y_{is} = \sum_{m=0}^K \beta_{m} \phi_m(s) +
\sum_{m=0}^K a_{im} \phi_m(s) + \epsilon\)</span>:</p>
<p>where</p>
<p><span class="math inline">\(y_{is}\)</span> is the phenotype for line
<span class="math inline">\(i\)</span> in environment <span
class="math inline">\(s\)</span>, <span
class="math inline">\(\phi_m(s)\)</span> is the Legendre polynomial of
order <span class="math inline">\(m\)</span> for environment <span
class="math inline">\(s\)</span>, <span
class="math inline">\(\beta_{m}\)</span> is the fixed effect of the
<span class="math inline">\(m^{th}\)</span>-order Legendre polynomial,
<span class="math inline">\(a_{im}\)</span> is the random additive
genetic value of the <span class="math inline">\(m^{th}\)</span>-order
Legendre polynomial, <span class="math inline">\(a_{im} \sim N(\mathbf
0, \mathbf G \otimes \mathbf \Sigma_a)\)</span> and <span
class="math inline">\(\mathbf \Sigma_r\)</span> is an <span
class="math inline">\(K \times K\)</span> covariance matrix. <span
class="math inline">\(\epsilon_{is}\)</span> is the residual value for
line <span class="math inline">\(i\)</span> in environment <span
class="math inline">\(s\)</span>, <span class="math inline">\(\epsilon
\sim N(O, \sigma^2_{\epsilon s})\)</span>.</p>
<p>After some testing, we chose <span
class="math inline">\(K=1\)</span>.</p>
<p>All these models (except for the random regression) were fitted in
the training set using a Bayesian approach as implemented in the <span
class="math inline">\(\sf BGLR\)</span> package. The random regression
model was fitted in the training set using a REML approach as
implemented in the <span class="math inline">\(\sf sommer\)</span>
package.</p>
<p>We used different cross-validation schemes:</p>
<ul>
<li><em>Random within environment scenario</em>. We assigned 17% of the
<em>lines</em> to the test set randomly and analyzed each
sex/temperature combination separately using the standard GBLUP model.
This procedure was repeated 6 times.</li>
<li><em>Random lines scenario</em>. We assigned 17% of the
<em>lines</em> (i.e., for the 2 sexes and the 3 temperatures) to the
test set randomly, and analyzed the sex/temperature combinations jointly
with the models above. The peculiarity of this scenario is that the
lines in the test set are not represented at all in the training set.
This procedure was repeated 6 times.</li>
<li><em>Random observations scenario</em>. We assigned 17% of the
<em>observations</em> (i.e., combination of line, sex and temperature)
to the test set randomly, and analyzed the sex/temperature combinations
jointly with the models above. The peculiarity of this scenario is that
all the lines, sexes and temperatures are represented in the training
set. This procedure was repeated 6 times.</li>
<li><em>New temperature and sex scenario</em>. We assigned all the
observations in a specific temperature and sex to the test set, i.e.,
that combination is never seen in the training set. This procedure was
repeated 6 times, once for each temperature/sex combination.</li>
</ul>
<p>Prediction accuracy was evaluated as the <span
class="math inline">\(R^2\)</span> of the regression of observed
phenotypes on predicted phenotypes in the test set.</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="random-within-environment-scenario" class="section level3">
<h3>Random within environment scenario</h3>
<p>This is the baseline scenario, where we acknowldege that the genetic
architecture of life span is different across environments and we
analyze them separately using G-BLUP.</p>
<pre class="r"><code>sex &lt;- c(1, 0)
temp &lt;- c(18,25,28)
model &lt;- &quot;G&quot;
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 6
n_row &lt;- length(reps) * length(sex) * length(temp) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;sex&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;r2&quot;, &quot;env&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+2, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;sex&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarEps&quot;, &quot;H2&quot;, &quot;env&quot;)


for(s in sex){
  for(t in temp){
    for(repp in reps){
      dat &lt;- readRDS(paste0(&quot;output/G_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_random_within_env_&quot;, s, &quot;_&quot;, t, &quot;_G_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
      
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- s
      res[i, 3] &lt;- t
      res[i, 4] &lt;- model
      res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      res[i, 6] &lt;- paste0(s, &quot;_&quot;, t)
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- s
      res_var[i, 3] &lt;- t
      res_var[i, 4] &lt;- model
      res_var[i, 5] &lt;- dat$model_fit$ETA$G$varU
      res_var[i, 6] &lt;- dat$model_fit$varE
      res_var[i, 7] &lt;- dat$model_fit$ETA$G$varU/(dat$model_fit$ETA$G$varU + dat$model_fit$varE)
      res_var[i, 8] &lt;- paste0(s, &quot;_&quot;, t)

    }
  }
}

#res_var_long &lt;- res_var %&gt;% select(rep, VarG, VarEps, env) %&gt;% gather(value=&quot;Var&quot;, key=&quot;Source&quot;, VarG, VarEps)
res_var &lt;- transform(res_var, env=as.factor(env))

p_var &lt;- ggplot(res_var, aes(x = env, y = H2)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(PVE)), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p_var)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20within%20env%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-orig-1">
Past versions of "random within env orig-1.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res &lt;- transform(res, env=as.factor(env))

p &lt;- ggplot(res, aes(x = env, y = r2)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(R)^2), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20within%20env%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-orig-2">
Past versions of "random within env orig-2.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(env) %&gt;% summarise(mean_r2=mean(r2), se_r2=(sd(r2)/length(reps))) %&gt;% as.data.frame()</code></pre>
<pre><code>   env    mean_r2       se_r2
1 0_18 0.02231063 0.003059148
2 0_25 0.11679648 0.013851290
3 0_28 0.03380754 0.005848980
4 1_18 0.07474225 0.009610133
5 1_25 0.05478486 0.009599134
6 1_28 0.02834367 0.003414872</code></pre>
<p>The results show that prediction accuracy varies across environments
and is generally low. As expected, scaling the phenotypes does not
affect accuracy in this case.</p>
</div>
<div id="random-lines-scenario" class="section level3">
<h3>Random lines scenario</h3>
<p>In this scenario, we combine the data across environments and fit the
different models.</p>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;,&#39;mvgblup&#39;,&#39;rrm&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- length(reps) * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      if(met == &quot;mvgblup&quot;){
        y_test &lt;- do.call(&quot;c&quot;, dat$y_test)
        yhat_test &lt;- do.call(&quot;c&quot;, dat$yhat_test)
        res[i, 4] &lt;- summary(lm(y_test ~ yhat_test))$r.squared
      } else {
        res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-orig-het-1">
Past versions of "random lines orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/ec76fc861d9f9cd7de140202a1a7952e36a829f4/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig het-1.png" target="_blank">ec76fc8</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/length(reps))) %&gt;% as.data.frame()</code></pre>
<pre><code>          model    mean_r2        se_r
1     E_het_var 0.73293872 0.004572708
2     G_het_var 0.01154011 0.001448275
3 GandE_het_var 0.74209476 0.004140810
4   GxE_het_var 0.74230381 0.003867658
5       mvgblup 0.74816815 0.004325620
6           rrm 0.42020671 0.011762662</code></pre>
<p>The results show that G-BLUP performs poorly. This is expected
because we are making the implicit assumption that genetic effects are
equal across environments, which is not the case. E-BLUP does very well
in this scenario because when combining the data across environments, a
lot of the variance is explained by the environmental effects, which we
have good power to estimate. GE-BLUP, GxE-BLUP, and mvG-BLUP give very
similar accuracy to E-BLUP. This is not surprising since some lines are
not in the training set, so the training-test transfer of information
happens mostly at the environment level. RRM does intermediately between
G-BLUP and the other models.</p>
<div id="gxe-by-chromosomes-models" class="section level4">
<h4>GxE by chromosomes models</h4>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- length(reps) * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-orig-het-chr-1">
Past versions of "random lines orig het chr-1.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-orig-het-chr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9b44f8d85c07220ce84ab313a001ab16bd8757c9/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig het chr-1.png" target="_blank">9b44f8d</a>
</td>
<td>
fmorgante
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/length(reps))) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2        se_r
1        GxE_het_var 0.7423038 0.003867658
2 GxE_by_chr_het_var 0.7408333 0.003681640</code></pre>
<p>Splitting the GxE component across chromosomes provides similar
results to the standard GxE-BLUP results.</p>
</div>
</div>
<div id="random-observations-scenario" class="section level3">
<h3>Random observations scenario</h3>
<p>In this scenario, every line and environment is potentially
represented in the training set. It is specific combinations of lines
and environments (e.g., line_101 in females at 18C) that are not seen in
the training set. This is the scenario where we expect an advantage from
using models with both genotype and environment information because the
training-test transfer of information happens at every level.</p>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;,&#39;mvgblup&#39;,&#39;rrm&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- length(reps) * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      if(met == &quot;mvgblup&quot;){
        y_test &lt;- do.call(&quot;c&quot;, dat$y_test)
        yhat_test &lt;- do.call(&quot;c&quot;, dat$yhat_test)
        res[i, 4] &lt;- summary(lm(y_test ~ yhat_test))$r.squared
      } else {
        res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-orig-het-1">
Past versions of "random obs orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/ec76fc861d9f9cd7de140202a1a7952e36a829f4/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig het-1.png" target="_blank">ec76fc8</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/length(reps))) %&gt;% as.data.frame()</code></pre>
<pre><code>          model    mean_r2        se_r
1     E_het_var 0.71554414 0.005081421
2     G_het_var 0.01406848 0.001210526
3 GandE_het_var 0.79661908 0.003645488
4   GxE_het_var 0.81897562 0.003028799
5       mvgblup 0.89112225 0.002393950
6           rrm 0.59850786 0.005146833</code></pre>
<p>The results show that, as expected, GE-BLUP improves prediction
accuracy over E-BLUP and GxE-BLUP achieves higher prediction accuracy
than GE-BLUP. mvG-BLUP achieves the highest accuracy as it can leverage
the strong genetic correlations among some environments, while also
accounting for genetic and environmental differences. RRM does
intermediately between G-BLUP and the other models.</p>
<div id="gxe-by-chromosomes-models-1" class="section level4">
<h4>GxE by chromosomes models</h4>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- length(reps) * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-orig-het-chr-1">
Past versions of "random obs orig het chr-1.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-orig-het-chr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9b44f8d85c07220ce84ab313a001ab16bd8757c9/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig het chr-1.png" target="_blank">9b44f8d</a>
</td>
<td>
fmorgante
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/length(reps))) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2        se_r
1        GxE_het_var 0.8189756 0.003028799
2 GxE_by_chr_het_var 0.8288852 0.002916905</code></pre>
<p>Splitting the GxE component across chromosomes provides slightly
better results than the standard GxE-BLUP results.</p>
</div>
</div>
<div id="new-temperature-and-sex-scenario" class="section level3">
<h3>New temperature and sex scenario</h3>
<p>In this scenario, we assigned all the observations in a specific
temperature and sex to the test set, i.e., that temperature/sex
combination is never seen in the training set.</p>
<pre class="r"><code>n_reps &lt;- 6
schemes &lt;- &quot;temp_sex&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;,&#39;rrm&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sex-orig-het-1">
Past versions of "sex orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-sex-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/ec76fc861d9f9cd7de140202a1a7952e36a829f4/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex orig het-1.png" target="_blank">ec76fc8</a>
</td>
<td>
fmorgante
</td>
<td>
2025-03-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model   mean_r2        se_r
1     E_het_var 0.0000000 0.000000000
2     G_het_var 0.2736952 0.004330556
3 GandE_het_var 0.3616820 0.010050622
4   GxE_het_var 0.4088251 0.006291057
5           rrm 0.3372179 0.020640791</code></pre>
<p>In general, prediction accuracy is lower in this scenario. As
expected, E-BLUP does not have any predictive ability; all the
observations in the test set get the same predicted phenotype because
they are equally “related” based on the environment. G-BLUP predicts
decently because all the lines are present in the test set. GE-BLUP
improves accuracy over G-BLUP because observations are now stratified by
both genetics and environment. GxE-BLUP does better than GE-BLUP because
it can capture the difference in genetic effects across environments and
“extrapolate” to a new environment better. RMM is now competitive with
GE-BLUP, although it produces a more variable performance.</p>
<div id="gxe-by-chromosomes-models-2" class="section level4">
<h4>GxE by chromosomes models</h4>
<pre class="r"><code>schemes &lt;- &quot;temp_sex&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sex-orig-het-chr-1">
Past versions of "sex orig het chr-1.png"
</button>
</p>
<div id="fig-no-spaces-sex-orig-het-chr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9b44f8d85c07220ce84ab313a001ab16bd8757c9/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex orig het chr-1.png" target="_blank">9b44f8d</a>
</td>
<td>
fmorgante
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2        se_r
1        GxE_het_var 0.4088251 0.006291057
2 GxE_by_chr_het_var 0.3684599 0.014506093</code></pre>
<p>Splitting the GxE component across chromosomes provides worse results
than the standard GxE-BLUP results. …</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.5 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/libs/gnu9/openblas/0.3.7/lib/libopenblasp-r0.3.7.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.4   tidyr_1.3.1   cowplot_1.1.3 ggplot2_3.5.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.13-1     pillar_1.9.0      compiler_4.1.2    bslib_0.8.0      
 [5] later_1.3.2       jquerylib_0.1.4   git2r_0.32.0      workflowr_1.7.0  
 [9] tools_4.1.2       digest_0.6.37     gtable_0.3.6      jsonlite_1.8.9   
[13] evaluate_1.0.1    lifecycle_1.0.4   tibble_3.2.1      pkgconfig_2.0.3  
[17] rlang_1.1.4       cli_3.6.2         rstudioapi_0.17.1 yaml_2.3.10      
[21] xfun_0.49         fastmap_1.2.0     withr_3.0.2       stringr_1.5.1    
[25] knitr_1.49        generics_0.1.3    fs_1.6.5          vctrs_0.6.5      
[29] sass_0.4.9        tidyselect_1.2.1  rprojroot_2.0.4   grid_4.1.2       
[33] glue_1.8.0        R6_2.6.1          fansi_1.0.6       rmarkdown_2.29   
[37] farver_2.1.2      purrr_1.0.2       magrittr_2.0.3    whisker_0.4.1    
[41] scales_1.3.0      promises_1.3.0    htmltools_0.5.8.1 colorspace_2.1-1 
[45] httpuv_1.6.11     labeling_0.4.3    utf8_1.2.4        stringi_1.8.4    
[49] munsell_0.5.1     cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
