<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Morgante" />

<meta name="date" content="2025-02-27" />

<title>Genomic prediction of life span with different models</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp_lifespan_gxe</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/morgante_lab/dgrp_lifespan_gxe">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Genomic prediction of life span with
different models</h1>
<h4 class="author">Fabio Morgante</h4>
<h4 class="date">February 27, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp_lifespan_gxe/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241004code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241004code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241004)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetreee4fa0f2bcb40cd22ea49909822db207dfc11a291targetblanke4fa0f2a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/e4fa0f2bcb40cd22ea49909822db207dfc11a291" target="_blank">e4fa0f2</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetreee4fa0f2bcb40cd22ea49909822db207dfc11a291targetblanke4fa0f2a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/e4fa0f2bcb40cd22ea49909822db207dfc11a291" target="_blank">e4fa0f2</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .snakemake/
    Ignored:    output/E_fit/
    Ignored:    output/E_het_var_fit/
    Ignored:    output/Et_fit/
    Ignored:    output/G_fit/
    Ignored:    output/G_het_var_fit/
    Ignored:    output/GandE_fit/
    Ignored:    output/GandE_het_var_fit/
    Ignored:    output/GxE_by_chr_fit/
    Ignored:    output/GxE_by_chr_het_var_fit/
    Ignored:    output/GxE_fit/
    Ignored:    output/GxE_gwas/
    Ignored:    output/GxE_gwas_fit/
    Ignored:    output/GxE_gwas_het_var_fit/
    Ignored:    output/GxE_het_var_fit/
    Ignored:    output/T_fit/
    Ignored:    output/TandE_fit/
    Ignored:    output/TxE_fit/
    Ignored:    output/test_sets/
    Ignored:    run/
    Ignored:    slurm/

Untracked files:
    Untracked:  code/fit_gwas_gxe_2envs_test.R
    Untracked:  code/fit_gwas_gxe_test.R
    Untracked:  scripts/run_fit_gwas_gxe_grammar_indicator.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_means.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_sex.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_temp.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lm.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_2envs_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_indicator.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_indicator_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_means.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_means_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_scaled_within_env_pheno_2envs_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_sex.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_sex_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_temp.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_temp_stand.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/dgrp_lifespan_gxe_genomic_results.Rmd</code>) and HTML
(<code>docs/dgrp_lifespan_gxe_genomic_results.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/e4fa0f2bcb40cd22ea49909822db207dfc11a291/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">e4fa0f2</a>
</td>
<td>
fmorgante
</td>
<td>
2025-02-27
</td>
<td>
Add GxE-GWAS and by chr results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/86e70e8f90276892cf04e441b9537e217c723826/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/52b983f29198fd75a765214854dc956166c9fecc/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">52b983f</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
<td>
Add het var results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/39ba942039ab6b2be5a263b35b5c3601920665d1/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">39ba942</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
<td>
Add new results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/9a93c45b800f43a4f4c6d14619221bf65971f03e/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">9a93c45</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/cc02d0d573816c5d58468d6311a982274f3be934/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">cc02d0d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
<td>
Add results of new temp/sex scenario
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/b7f69008efb9bf3dff0435a11b27e2f5107ce07e/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">b7f6900</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/d4fe1263e221619e0dc9c12c7c9b6c2d2a7b5096/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">d4fe126</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Add descriptions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/a108f2043fb03f898a50d62f39fe6c00f21db686/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">a108f20</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
<td>
Add results with different scaling of y
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/5247c8b6f6b9e21e549da947eb11e5513695abc2/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">5247c8b</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/66afcdd827caf63326608f0f2a09dc2f0ea3a2c3/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">66afcdd</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/f49ca0a50356e06fa5b2d8c0531eeef0980fe4ea/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">f49ca0a</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-24
</td>
<td>
Add across tempreature results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/bfc1461a735460ba936cfd01ccd96fccd70a08b3/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">bfc1461</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/db9c383061548e3043f9741fe822a0c3d6725f12/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">db9c383</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/1091563f08d32456638eee023d61a2ecf09576ef/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">1091563</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Minor fixes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/dce17d9d41f5075cccb08dd7584467fb186b8a85/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">dce17d9</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9558bc7af36bf7ad97e5faeeecc24a5ba7e87246/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">9558bc7</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Use R2 instead of r in genomic results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/8530bb379906a619955b9046e9d54820b473503a/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">8530bb3</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/7aecac219990c05ac11be6910c7e0f3a97f3667b/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">7aecac2</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-15
</td>
<td>
Add PVE to genomic results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/77678919a17a358a8186f04be2ae3688ee0d1d70/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">7767891</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/dgrp_lifespan_gxe_genomic_results.html" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/24fb66c0b6810d35bbd8eb641186407bf7d2abd0/analysis/dgrp_lifespan_gxe_genomic_results.Rmd" target="_blank">24fb66c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Update results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>###Load libraries
library(ggplot2)
library(cowplot)
library(tidyr)
library(dplyr)

prefix &lt;- &quot;dgrp_lifespan_gxe&quot;
n_reps &lt;- 20
pvals &lt;- c(&quot;all&quot;, &quot;1e-3&quot;, &quot;1e-5&quot;, &quot;1e-8&quot;, &quot;1e-10&quot;, &quot;1e-12&quot;, &quot;1e-14&quot;)</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The goal of this analysis is to assess the performance of the linear
mixed models including genotype only, environment only, genotype +
environment, genotype + environment + genotype <span
class="math inline">\(\times\)</span> environment information at
predicting life span in the DGRP.</p>
<p>We used data from <a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000645">Huang
et al. (2020)</a>, which has life span measurements for <span
class="math inline">\(n=176\)</span> lines at 3 temperatures (18C, 25C,
28C) for the 2 sexes (coded as 1=females and 0=males), which results in
<span class="math inline">\(q=1,056\)</span> observations. The authors
showed that there is extensive <span class="math inline">\(G \times
E\)</span> affecting life span. In the present work, we used both
temperature and sex as <span class="math inline">\(c=2\)</span>
environmental variables, for a total of <span
class="math inline">\(r=6\)</span> environments.</p>
<p>We fitted the following linear mixed models:</p>
<ul>
<li>G-BLUP – <span class="math inline">\(y = \mu + g +
\epsilon\)</span>.</li>
<li>E-BLUP – <span class="math inline">\(y = \mu + e +
\epsilon\)</span>.</li>
<li>GE-BLUP – <span class="math inline">\(y = \mu + g + e +
\epsilon\)</span>.</li>
<li>GxE-BLUP – <span class="math inline">\(y = \mu + g + e + ge +
\epsilon\)</span>.</li>
</ul>
<p>where</p>
<p><span class="math inline">\(y\)</span> is a <span
class="math inline">\(q\)</span>-vector of phenotypic observations,
<span class="math inline">\(\mu\)</span> is a <span
class="math inline">\(q\)</span>-vector of intercept values, <span
class="math inline">\(g\)</span> is a <span
class="math inline">\(q\)</span>-vector of additive genetic values,
<span class="math inline">\(g \sim N_q(\mathbf O, \mathbf{ZGZ^\intercal}
\sigma^2_g)\)</span>, <span class="math inline">\(\mathbf Z\)</span> is
a <span class="math inline">\(q \times n\)</span> incidence matrix,
<span class="math inline">\(\mathbf G\)</span> is a <span
class="math inline">\(n \times n\)</span> genomic relationship matrix
(GRM), <span class="math inline">\(e\)</span> is a <span
class="math inline">\(q\)</span>-vector of environmental values, <span
class="math inline">\(e \sim N_q(\mathbf O, \mathbf{E}
\sigma^2_e)\)</span>, <span class="math inline">\(\mathbf E\)</span> is
a <span class="math inline">\(q \times q\)</span> matrix of similarity
based on environmental variables, computed as <span
class="math inline">\(\mathbf E = \mathbf{XX}^\intercal\)</span>, <span
class="math inline">\(\mathbf X\)</span> is a <span
class="math inline">\(q \times c\)</span> matrix of environmental
measurements, <span class="math inline">\(ge \sim N_q(\mathbf O,
\mathbf{ZGZ^\intercal \circ E} \sigma^2_{ge})\)</span>, <span
class="math inline">\(\epsilon\)</span> is a <span
class="math inline">\(q\)</span>-vector of residual values, <span
class="math inline">\(\epsilon \sim N_q(\mathbf O, \mathbf{I}
\sigma^2_\epsilon)\)</span>.</p>
<p>These models were fitted in the training set using a Bayesian
approach as implemented in the <span class="math inline">\(\sf
BGLR\)</span> package.</p>
<p>We used different cross-validation schemes:</p>
<ul>
<li><em>Random within environment scenario</em>. We assigned 20% of the
<em>lines</em> to the test set randomly and analyzed each
sex/temperature combination separately using the standard GBLUP model.
This procedure was repeated 20 times.</li>
<li><em>Random lines scenario</em>. We assigned 20% of the
<em>lines</em> (i.e., for the 2 sexes and the 3 temperatures) to the
test set randomly, and analyzed the sex/temperature combinations jointly
with the models above. The peculiarity of this scenario is that the
lines in the test set are not represented at all in the training set.
This procedure was repeated 20 times.</li>
<li><em>Random observations scenario</em>. We assigned 20% of the
<em>observations</em> (i.e., combination of line, sex and temperature)
to the test set randomly, and analyzed the sex/temperature combinations
jointly with the models above. The peculiarity of this scenario is that
all the lines, sexes and temperatures are represented in the training
set. This procedure was repeated 20 times.</li>
<li><em>New temperature scenario</em>. We assigned all the observations
in a specific temperature to the test set, i.e., that temperature is
never seen in the training set. This procedure was repeated 3 times,
once for each temperature.</li>
<li><em>New temperature and sex scenario</em>. We assigned all the
observations in a specific temperature and sex to the test set, i.e.,
that combination is never seen in the training set. This procedure was
repeated 6 times, once for each temperature/sex combination.</li>
</ul>
<p>We used different treatments of the phenotype:</p>
<ul>
<li>Leave it on the original scale.</li>
<li>Scale it within each sex/temperature combination.</li>
</ul>
<p>Prediction accuracy was evaluated as the <span
class="math inline">\(R^2\)</span> of the regression of observed
phenotypes on predicted phenotypes in the test set.</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="random-within-environment-scenario" class="section level3">
<h3>Random within environment scenario</h3>
<p>This is the baseline scenario, where we acknowldege that the genetic
architecture of life span is different across environments and we
analyze them separately using G-BLUP.</p>
<div id="standard-models" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>sex &lt;- c(1, 0)
temp &lt;- c(18,25,28)
model &lt;- &quot;G&quot;
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 6
n_row &lt;- n_reps * length(sex) * length(temp) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;sex&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;r2&quot;, &quot;env&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+2, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;sex&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarEps&quot;, &quot;H2&quot;, &quot;env&quot;)


for(s in sex){
  for(t in temp){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/G_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_random_within_env_&quot;, s, &quot;_&quot;, t, &quot;_G_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
      
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- s
      res[i, 3] &lt;- t
      res[i, 4] &lt;- model
      res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      res[i, 6] &lt;- paste0(s, &quot;_&quot;, t)
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- s
      res_var[i, 3] &lt;- t
      res_var[i, 4] &lt;- model
      res_var[i, 5] &lt;- dat$model_fit$ETA$G$varU
      res_var[i, 6] &lt;- dat$model_fit$varE
      res_var[i, 7] &lt;- dat$model_fit$ETA$G$varU/(dat$model_fit$ETA$G$varU + dat$model_fit$varE)
      res_var[i, 8] &lt;- paste0(s, &quot;_&quot;, t)

    }
  }
}

#res_var_long &lt;- res_var %&gt;% select(rep, VarG, VarEps, env) %&gt;% gather(value=&quot;Var&quot;, key=&quot;Source&quot;, VarG, VarEps)
res_var &lt;- transform(res_var, env=as.factor(env))

p_var &lt;- ggplot(res_var, aes(x = env, y = H2)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(PVE)), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p_var)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20within%20env%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-orig-1">
Past versions of "random within env orig-1.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res &lt;- transform(res, env=as.factor(env))

p &lt;- ggplot(res, aes(x = env, y = r2)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(R)^2), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20within%20env%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-orig-2">
Past versions of "random within env orig-2.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env orig-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(env) %&gt;% summarise(mean_r2=mean(r2), se_r2=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>   env     mean_r2        se_r2
1 0_18 0.003732303 0.0002240377
2 0_25 0.054900465 0.0013920539
3 0_28 0.008589620 0.0006014524
4 1_18 0.033372220 0.0009984413
5 1_25 0.050655298 0.0016030720
6 1_28 0.006023807 0.0003804171</code></pre>
</div>
<div id="phenotype-scaled-within-environment" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>sex &lt;- c(1, 0)
temp &lt;- c(18,25,28)
model &lt;- &quot;G&quot;
pheno &lt;- &quot;scaled_within_env_pheno&quot;


i &lt;- 0

n_col &lt;- 6
n_row &lt;- n_reps * length(sex) * length(temp) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;sex&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;r2&quot;, &quot;env&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+2, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;sex&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarEps&quot;, &quot;H2&quot;, &quot;env&quot;)


for(s in sex){
  for(t in temp){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/G_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_random_within_env_&quot;, s, &quot;_&quot;, t, &quot;_G_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
      
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- s
      res[i, 3] &lt;- t
      res[i, 4] &lt;- model
      res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      res[i, 6] &lt;- paste0(s, &quot;_&quot;, t)
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- s
      res_var[i, 3] &lt;- t
      res_var[i, 4] &lt;- model
      res_var[i, 5] &lt;- dat$model_fit$ETA$G$varU
      res_var[i, 6] &lt;- dat$model_fit$varE
      res_var[i, 7] &lt;- dat$model_fit$ETA$G$varU/(dat$model_fit$ETA$G$varU + dat$model_fit$varE)
      res_var[i, 8] &lt;- paste0(s, &quot;_&quot;, t)

    }
  }
}

#res_var_long &lt;- res_var %&gt;% select(rep, VarG, VarEps, env) %&gt;% gather(value=&quot;Var&quot;, key=&quot;Source&quot;, VarG, VarEps)
res_var &lt;- transform(res_var, env=as.factor(env))

p_var &lt;- ggplot(res_var, aes(x = env, y = H2)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(PVE)), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p_var)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20within%20env%20scaled%20within%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-scaled-within-env-1">
Past versions of "random within env scaled within env-1.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-scaled-within-env-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env scaled within env-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env scaled within env-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res &lt;- transform(res, env=as.factor(env))

p &lt;- ggplot(res, aes(x = env, y = r2)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(R)^2), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20within%20env%20scaled%20within%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-scaled-within-env-2">
Past versions of "random within env scaled within env-2.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-scaled-within-env-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env scaled within env-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random within env scaled within env-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(env) %&gt;% summarise(mean_r2=mean(r2), se_r2=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>   env     mean_r2        se_r2
1 0_18 0.003732303 0.0002240377
2 0_25 0.054900465 0.0013920539
3 0_28 0.008589620 0.0006014524
4 1_18 0.033372220 0.0009984413
5 1_25 0.050655298 0.0016030720
6 1_28 0.006023807 0.0003804171</code></pre>
<p>The results show that prediction accuracy varies across environments
and is generally low. As expected, scaling the phenotypes does not
affect accuracy in this case.</p>
</div>
</div>
</div>
</div>
<div id="random-lines-scenario" class="section level3">
<h3>Random lines scenario</h3>
<p>In this scenario, we combine the data across environments and fit the
different models.</p>
<div id="standard-models-1" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance-1" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-1" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-orig-1">
Past versions of "random lines orig-1.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-orig-2">
Past versions of "random lines orig-2.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model     mean_r2         se_r
1     E 0.723562228 0.0006686715
2     G 0.004755356 0.0001447866
3 GandE 0.726580957 0.0006751025
4   GxE 0.726106729 0.0006674381</code></pre>
</div>
<div id="phenotype-scaled-within-environment-1" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;scaled_within_env_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20within%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-within-env-1">
Past versions of "random lines within env-1.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-within-env-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines within env-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines within env-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20within%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-within-env-2">
Past versions of "random lines within env-2.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-within-env-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines within env-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines within env-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model     mean_r2         se_r
1     E 0.001046068 4.978764e-05
2     G 0.019482127 6.792342e-04
3 GandE 0.018358483 6.792282e-04
4   GxE 0.019226188 6.412072e-04</code></pre>
<p>With the phenotype on the original scale, the results show that
G-BLUP performs poorly. This is expected because we are making the
implicit assumption that genetic effects are equal across environments,
which is not the case. E-BLUP does very well in this scenario because
when combining the data across environments, a lot of the variance is
explained by the environmental effects, which we have good power to
estimate. GE-BLUP and GxE-BLUP give very similar accuracy to E-BLUP.
This is not surprising since some lines are not in the training set, so
the training-test transfer of information happens mostly at the
environment level.</p>
<p>However, with the phenotype scaled within environment, E-BLUP now
does not predict at all because we have zeroed out differences between
environments. All the models including G do pretty poorly.</p>
</div>
</div>
<div id="heterogeneous-residual-variance" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-2" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-orig-het-1">
Past versions of "random lines orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random lines orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model     mean_r2         se_r
1     E_het_var 0.723426379 0.0006849420
2     G_het_var 0.004695946 0.0001691966
3 GandE_het_var 0.726939663 0.0007040385
4   GxE_het_var 0.725793527 0.0006896233</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-gwas-informed-models" class="section level4">
<h4>GxE GWAS-informed models</h4>
<div id="homogeneous-residual-variance-2" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-3" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- &#39;GxE_gwas&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>     model  pval   mean_r2         se_r
1 GxE_gwas   all 0.7261067 0.0006674381
2 GxE_gwas  1e-3 0.7252206 0.0006833508
3 GxE_gwas  1e-5 0.7249580 0.0006784849
4 GxE_gwas  1e-8 0.7240402 0.0006630034
5 GxE_gwas 1e-10 0.7219400 0.0006896685
6 GxE_gwas 1e-12 0.7171943 0.0007379588
7 GxE_gwas 1e-14 0.7088334 0.0007259222</code></pre>
<p>…</p>
</div>
</div>
<div id="heterogeneous-residual-variance-1" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-4" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- &#39;GxE_gwas_het_var&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20het%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>             model  pval   mean_r2         se_r
1 GxE_gwas_het_var   all 0.7261067 0.0006674381
2 GxE_gwas_het_var  1e-3 0.7251122 0.0006888704
3 GxE_gwas_het_var  1e-5 0.7248842 0.0006931091
4 GxE_gwas_het_var  1e-8 0.7239326 0.0006744185
5 GxE_gwas_het_var 1e-10 0.7220142 0.0006897480
6 GxE_gwas_het_var 1e-12 0.7175195 0.0007247041
7 GxE_gwas_het_var 1e-14 0.7093227 0.0007258372</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-by-chromosomes-models" class="section level4">
<h4>GxE by chromosomes models</h4>
<div id="homogeneous-residual-variance-3" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-5" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&quot;GxE&quot;, &quot;GxE_by_chr&quot;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_1xE&quot;, &quot;VarG_2xE&quot;, &quot;VarG_3xE&quot;, 
                       &quot;VarG_4xE&quot;, &quot;VarG_5xE&quot;, &quot;VarG_6xE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] &lt;- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] &lt;- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] &lt;- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] &lt;- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] &lt;- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] &lt;- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] &lt;- dat$model_fit$varE
      res_var[i, 14] &lt;- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propG_1xE &lt;- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE &lt;- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE &lt;- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE &lt;- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE &lt;- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE &lt;- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propG_1xE&quot;, &quot;propG_2xE&quot;, &quot;propG_3xE&quot;, 
                                                &quot;propG_4xE&quot;, &quot;propG_5xE&quot;, &quot;propG_6xE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), 
                               labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_2LxE&quot;, &quot;VarG_2RxE&quot;, &quot;VarG_3LxE&quot;, 
                                        &quot;VarG_3RxE&quot;, &quot;VarG_XxE&quot;, &quot;VarG_4xE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  #scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 120 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 120 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20chr-2.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>       model   mean_r2         se_r
1        GxE 0.7261067 0.0006674381
2 GxE_by_chr 0.7216409 0.0007520302</code></pre>
</div>
</div>
<div id="heterogeneous-residual-variance-2" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-6" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20lines%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2         se_r
1        GxE_het_var 0.7257935 0.0006896233
2 GxE_by_chr_het_var 0.7227647 0.0007641816</code></pre>
</div>
</div>
</div>
</div>
<div id="random-observations-scenario" class="section level3">
<h3>Random observations scenario</h3>
<p>In this scenario, every line and environment is potentially
represented in the training set. It is specific combinations of lines
and environments (e.g., line_101 in females at 18C) that are not seen in
the training set. This is the scenario where we expect an advantage from
using models with both genotype and environment information because the
training-test transfer of information happens at every level.</p>
<div id="standard-models-2" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance-4" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-7" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-orig-1">
Past versions of "random obs orig-1.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-orig-2">
Past versions of "random obs orig-2.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model     mean_r2         se_r
1     E 0.722552163 0.0007951532
2     G 0.001119255 0.0001024137
3 GandE 0.786877061 0.0006237049
4   GxE 0.802392982 0.0008092222</code></pre>
</div>
<div id="phenotype-scaled-within-environment-2" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;scaled_within_env_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 100 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20within%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-within-env-1">
Past versions of "random obs within env-1.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-within-env-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs within env-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs within env-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20within%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-within-env-2">
Past versions of "random obs within env-2.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-within-env-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs within env-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs within env-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model     mean_r2         se_r
1     E 0.001964095 9.874971e-05
2     G 0.342309928 1.143042e-03
3 GandE 0.340874524 1.131047e-03
4   GxE 0.358846080 1.144672e-03</code></pre>
<p>With the phenotype on the original scale, the results show that, as
expected, GE-BLUP improves prediction accuracy over E-BLUP. GxE-BLUP
achieves the highest prediction accuracy.</p>
<p>However, with the phenotype scaled within environment, E-BLUP now
does not predict at all because we have zeroed out differences between
environments. G-BLUP and GE-BLUP perform equally well (why does G-BLUP
work decently now?) and GxE-BLUP achieves the highest prediction
accuracy.</p>
</div>
</div>
<div id="heterogeneous-residual-variance-3" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-8" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-orig-het-1">
Past versions of "random obs orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/random obs orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model     mean_r2         se_r
1     E_het_var 0.721997508 0.0007902213
2     G_het_var 0.002822643 0.0001601667
3 GandE_het_var 0.785036894 0.0007417887
4   GxE_het_var 0.794021024 0.0008231949</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-gwas-informed-models-1" class="section level4">
<h4>GxE GWAS-informed models</h4>
<div id="homogeneous-residual-variance-5" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-9" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- &#39;GxE_gwas&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<pre><code>Warning: Removed 16 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 16 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>     model  pval   mean_r2         se_r
1 GxE_gwas   all 0.8023930 0.0008092222
2 GxE_gwas  1e-3 0.8006372 0.0008156830
3 GxE_gwas  1e-5 0.8001850 0.0008224507
4 GxE_gwas  1e-8 0.7982902 0.0008250728
5 GxE_gwas 1e-10 0.8005223 0.0008624407
6 GxE_gwas 1e-12 0.7993103 0.0007738799
7 GxE_gwas 1e-14 0.8037594 0.0002967296</code></pre>
<p>…</p>
</div>
</div>
<div id="heterogeneous-residual-variance-4" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-10" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- &#39;GxE_gwas_het_var&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<pre><code>Warning: Removed 16 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 16 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20het%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>             model  pval   mean_r2         se_r
1 GxE_gwas_het_var   all 0.8023930 0.0008092222
2 GxE_gwas_het_var  1e-3 0.7926826 0.0008706039
3 GxE_gwas_het_var  1e-5 0.7924163 0.0008754583
4 GxE_gwas_het_var  1e-8 0.7899405 0.0008635915
5 GxE_gwas_het_var 1e-10 0.7887702 0.0009058693
6 GxE_gwas_het_var 1e-12 0.7877377 0.0008455574
7 GxE_gwas_het_var 1e-14 0.7941286 0.0003974058</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-by-chromosomes-models-1" class="section level4">
<h4>GxE by chromosomes models</h4>
<div id="homogeneous-residual-variance-6" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-11" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&quot;GxE&quot;, &quot;GxE_by_chr&quot;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_1xE&quot;, &quot;VarG_2xE&quot;, &quot;VarG_3xE&quot;, 
                       &quot;VarG_4xE&quot;, &quot;VarG_5xE&quot;, &quot;VarG_6xE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] &lt;- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] &lt;- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] &lt;- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] &lt;- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] &lt;- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] &lt;- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] &lt;- dat$model_fit$varE
      res_var[i, 14] &lt;- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propG_1xE &lt;- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE &lt;- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE &lt;- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE &lt;- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE &lt;- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE &lt;- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propG_1xE&quot;, &quot;propG_2xE&quot;, &quot;propG_3xE&quot;, 
                                                &quot;propG_4xE&quot;, &quot;propG_5xE&quot;, &quot;propG_6xE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), 
                               labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_2LxE&quot;, &quot;VarG_2RxE&quot;, &quot;VarG_3LxE&quot;, 
                                        &quot;VarG_3RxE&quot;, &quot;VarG_XxE&quot;, &quot;VarG_4xE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  #scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 120 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 120 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20chr-2.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>       model   mean_r2         se_r
1        GxE 0.8023930 0.0008092222
2 GxE_by_chr 0.7919958 0.0008315590</code></pre>
</div>
</div>
<div id="heterogeneous-residual-variance-5" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-12" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/random%20obs%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2         se_r
1        GxE_het_var 0.7940210 0.0008231949
2 GxE_by_chr_het_var 0.7773769 0.0009459000</code></pre>
</div>
</div>
</div>
</div>
<div id="new-temperature-scenario" class="section level3">
<h3>New temperature scenario</h3>
<p>In this scenario, we assigned all the observations in a specific
temperature to the test set, i.e., that temperature is never seen in the
training set. Thus, there is no transfer of training-test information at
the temperature level, which is the factor that explains the majority of
the phenotypic variance.</p>
<div id="standard-models-3" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance-7" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-13" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 3
schemes &lt;- &quot;temp&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(temp, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 15 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 15 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-temp-orig-1">
Past versions of "across temp orig-1.png"
</button>
</p>
<div id="fig-no-spaces-across-temp-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp orig-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-temp-orig-2">
Past versions of "across temp orig-2.png"
</button>
</p>
<div id="fig-no-spaces-across-temp-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp orig-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model    mean_r2        se_r
1     E 0.01530178 0.004543686
2     G 0.27412654 0.010361872
3 GandE 0.28817305 0.018338645
4   GxE 0.26904402 0.043655817</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>   temp scheme model          r2
1     1   temp     G 0.277155191
2     2   temp     G 0.303586981
3     3   temp     G 0.241637452
4     1   temp     E 0.027398587
5     2   temp     E 0.017974397
6     3   temp     E 0.000532356
7     1   temp GandE 0.300870951
8     2   temp GandE 0.335729817
9     3   temp GandE 0.227918395
10    1   temp   GxE 0.305878684
11    2   temp   GxE 0.377649823
12    3   temp   GxE 0.123603541</code></pre>
</div>
<div id="phenotype-scaled-within-environment-3" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>n_reps &lt;- 3
schemes &lt;- &quot;temp&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;scaled_within_env_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(temp, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 15 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 15 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20within%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-within-env-1">
Past versions of "across within env-1.png"
</button>
</p>
<div id="fig-no-spaces-across-within-env-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across within env-1.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20within%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-within-env-2">
Past versions of "across within env-2.png"
</button>
</p>
<div id="fig-no-spaces-across-within-env-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/970178d00a7a639b591604e6c2f4ca5bacc03154/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across within env-2.png" target="_blank">970178d</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model      mean_r2         se_r
1     E 3.454929e-22 1.573713e-22
2     G 3.050513e-01 1.508132e-02
3 GandE 3.050231e-01 1.493395e-02
4   GxE 3.072790e-01 1.858227e-02</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>   temp scheme model           r2
1     1   temp     G 2.834616e-01
2     2   temp     G 3.570460e-01
3     3   temp     G 2.746462e-01
4     1   temp     E 9.561634e-23
5     2   temp     E 8.900294e-22
6     3   temp     E 5.083288e-23
7     1   temp GandE 2.832124e-01
8     2   temp GandE 3.565538e-01
9     3   temp GandE 2.753030e-01
10    1   temp   GxE 2.946487e-01
11    2   temp   GxE 3.682574e-01
12    3   temp   GxE 2.589311e-01</code></pre>
<p>With the phenotype on the original scale, in general, prediction
accuracy is lower in this scenario. As expected, E-BLUP predicts poorly.
G-BLUP predicts decently because all the lines are present in the test
set. GE-BLUP improves accuracy over G-BLUP. GxE-BLUP is worse than
G-BLUP and GE-BLUP, on average. However, when looking at the
temperature-specific accuracies, we can see that it is only the
prediction of lifespan at 28C that has lower accuracy.</p>
<p>With the phenotype scaled within environment, the pattern is very
similar in this scenario.</p>
</div>
</div>
<div id="heterogeneous-residual-variance-6" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-14" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 3
schemes &lt;- &quot;temp&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-temp-orig-het-1">
Past versions of "across temp orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-across-temp-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model    mean_r2        se_r
1     E_het_var 0.01530178 0.004543686
2     G_het_var 0.23329755 0.015981306
3 GandE_het_var 0.26323413 0.010625486
4   GxE_het_var 0.27972420 0.023458242</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-gwas-informed-models-2" class="section level4">
<h4>GxE GWAS-informed models</h4>
<div id="homogeneous-residual-variance-8" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-15" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 3
schemes &lt;- &quot;temp&quot;
model &lt;- &#39;GxE_gwas&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>     model  pval   mean_r2       se_r
1 GxE_gwas   all 0.2690440 0.04365582
2 GxE_gwas  1e-3 0.2098978 0.05227379
3 GxE_gwas  1e-5 0.2085205 0.05288181
4 GxE_gwas  1e-8 0.2218150 0.05213748
5 GxE_gwas 1e-10 0.2422868 0.04560523
6 GxE_gwas 1e-12 0.2539924 0.04385491
7 GxE_gwas 1e-14 0.3784992         NA</code></pre>
<p>…</p>
</div>
</div>
<div id="heterogeneous-residual-variance-7" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-16" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 3
schemes &lt;- &quot;temp&quot;
model &lt;- &#39;GxE_gwas_het_var&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig%20het%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>             model  pval   mean_r2       se_r
1 GxE_gwas_het_var   all 0.2690440 0.04365582
2 GxE_gwas_het_var  1e-3 0.2260167 0.04796158
3 GxE_gwas_het_var  1e-5 0.2177884 0.04915410
4 GxE_gwas_het_var  1e-8 0.2307550 0.04805276
5 GxE_gwas_het_var 1e-10 0.2460663 0.04249919
6 GxE_gwas_het_var 1e-12 0.2647197 0.03244546
7 GxE_gwas_het_var 1e-14 0.3588890         NA</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-by-chromosomes-models-2" class="section level4">
<h4>GxE by chromosomes models</h4>
<div id="homogeneous-residual-variance-9" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-17" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;temp&quot;
model &lt;- c(&quot;GxE&quot;, &quot;GxE_by_chr&quot;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_1xE&quot;, &quot;VarG_2xE&quot;, &quot;VarG_3xE&quot;, 
                       &quot;VarG_4xE&quot;, &quot;VarG_5xE&quot;, &quot;VarG_6xE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] &lt;- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] &lt;- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] &lt;- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] &lt;- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] &lt;- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] &lt;- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] &lt;- dat$model_fit$varE
      res_var[i, 14] &lt;- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propG_1xE &lt;- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE &lt;- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE &lt;- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE &lt;- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE &lt;- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE &lt;- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propG_1xE&quot;, &quot;propG_2xE&quot;, &quot;propG_3xE&quot;, 
                                                &quot;propG_4xE&quot;, &quot;propG_5xE&quot;, &quot;propG_6xE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), 
                               labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_2LxE&quot;, &quot;VarG_2RxE&quot;, &quot;VarG_3LxE&quot;, 
                                        &quot;VarG_3RxE&quot;, &quot;VarG_XxE&quot;, &quot;VarG_4xE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  #scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 18 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 18 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig%20chr-2.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>       model   mean_r2       se_r
1        GxE 0.2690440 0.04365582
2 GxE_by_chr 0.2572963 0.04472022</code></pre>
</div>
</div>
<div id="heterogeneous-residual-variance-8" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-18" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;temp&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2       se_r
1        GxE_het_var 0.2797242 0.02345824
2 GxE_by_chr_het_var 0.2444803 0.04281253</code></pre>
</div>
</div>
</div>
</div>
<div id="new-sex-scenario" class="section level3">
<h3>New sex scenario</h3>
<p>In this scenario, we assigned all the observations in a specific sex
to the test set, i.e., that sex is never seen in the training set. Thus,
there is no transfer of training-test information at the sex level.</p>
<div id="standard-models-4" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance-10" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-19" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;sex&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;sex&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;sex&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(sex, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-sex-orig-1">
Past versions of "across sex orig-1.png"
</button>
</p>
<div id="fig-no-spaces-across-sex-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across sex orig-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-sex-orig-2">
Past versions of "across sex orig-2.png"
</button>
</p>
<div id="fig-no-spaces-across-sex-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across sex orig-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model    mean_r2        se_r
1     E 0.72359620 0.011658860
2     G 0.09479391 0.003661257
3 GandE 0.81827983 0.009046977
4   GxE 0.73233395 0.009450426</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>  sex scheme model         r2
1   1    sex     G 0.09997171
2   2    sex     G 0.08961611
3   1    sex     E 0.70710808
4   2    sex     E 0.74008431
5   1    sex GandE 0.80548547
6   2    sex GandE 0.83107419
7   1    sex   GxE 0.74569887
8   2    sex   GxE 0.71896903</code></pre>
</div>
<div id="phenotype-scaled-within-environment-4" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;sex&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;scaled_within_env_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;sex&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;sex&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(sex, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-sex-env-1">
Past versions of "across sex env-1.png"
</button>
</p>
<div id="fig-no-spaces-across-sex-env-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across sex env-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-sex-env-2">
Past versions of "across sex env-2.png"
</button>
</p>
<div id="fig-no-spaces-across-sex-env-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across sex env-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model      mean_r2         se_r
1     E 4.183670e-23 2.840449e-23
2     G 4.285604e-01 7.043875e-03
3 GandE 4.282807e-01 7.565472e-03
4   GxE 4.925058e-01 1.064954e-02</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>  sex scheme model           r2
1   1    sex     G 4.185989e-01
2   2    sex     G 4.385219e-01
3   1    sex     E 8.200672e-23
4   2    sex     E 1.666677e-24
5   1    sex GandE 4.175815e-01
6   2    sex GandE 4.389799e-01
7   1    sex   GxE 4.774451e-01
8   2    sex   GxE 5.075665e-01</code></pre>
<p>With the phenotype on the original scale, …</p>
<p>With the phenotype scaled within environment, …</p>
</div>
</div>
<div id="heterogeneous-residual-variance-9" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-20" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;sex&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-across-sex-orig-het-1">
Past versions of "across sex orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-across-sex-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across sex orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model    mean_r2        se_r
1     E_het_var 0.72359620 0.011658860
2     G_het_var 0.06436602 0.004608881
3 GandE_het_var 0.79402164 0.006638155
4   GxE_het_var 0.79282122 0.009319443</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-gwas-informed-models-3" class="section level4">
<h4>GxE GWAS-informed models</h4>
<div id="homogeneous-residual-variance-11" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-21" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;sex&quot;
model &lt;- &#39;GxE_gwas&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>     model  pval   mean_r2         se_r
1 GxE_gwas   all 0.7323339 0.0094504263
2 GxE_gwas  1e-3 0.6059428 0.0006163993
3 GxE_gwas  1e-5 0.6255098 0.0044034258
4 GxE_gwas  1e-8 0.6951467 0.0081814058
5 GxE_gwas 1e-10 0.7180181 0.0063268675
6 GxE_gwas 1e-12       NaN           NA
7 GxE_gwas 1e-14       NaN           NA</code></pre>
<p>…</p>
</div>
</div>
<div id="heterogeneous-residual-variance-10" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-22" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;sex&quot;
model &lt;- &#39;GxE_gwas_het_var&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig%20het%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>             model  pval   mean_r2         se_r
1 GxE_gwas_het_var   all 0.7323339 0.0094504263
2 GxE_gwas_het_var  1e-3 0.6367460 0.0014540747
3 GxE_gwas_het_var  1e-5 0.6315222 0.0008584284
4 GxE_gwas_het_var  1e-8 0.6827499 0.0034841792
5 GxE_gwas_het_var 1e-10 0.7608040 0.0096147196
6 GxE_gwas_het_var 1e-12       NaN           NA
7 GxE_gwas_het_var 1e-14       NaN           NA</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-by-chromosomes-models-3" class="section level4">
<h4>GxE by chromosomes models</h4>
<div id="homogeneous-residual-variance-12" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-23" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;sex&quot;
model &lt;- c(&quot;GxE&quot;, &quot;GxE_by_chr&quot;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_1xE&quot;, &quot;VarG_2xE&quot;, &quot;VarG_3xE&quot;, 
                       &quot;VarG_4xE&quot;, &quot;VarG_5xE&quot;, &quot;VarG_6xE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] &lt;- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] &lt;- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] &lt;- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] &lt;- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] &lt;- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] &lt;- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] &lt;- dat$model_fit$varE
      res_var[i, 14] &lt;- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propG_1xE &lt;- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE &lt;- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE &lt;- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE &lt;- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE &lt;- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE &lt;- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propG_1xE&quot;, &quot;propG_2xE&quot;, &quot;propG_3xE&quot;, 
                                                &quot;propG_4xE&quot;, &quot;propG_5xE&quot;, &quot;propG_6xE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), 
                               labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_2LxE&quot;, &quot;VarG_2RxE&quot;, &quot;VarG_3LxE&quot;, 
                                        &quot;VarG_3RxE&quot;, &quot;VarG_XxE&quot;, &quot;VarG_4xE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  #scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 12 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 12 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig%20chr-2.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>       model   mean_r2        se_r
1        GxE 0.7323339 0.009450426
2 GxE_by_chr 0.6430582 0.006419445</code></pre>
</div>
</div>
<div id="heterogeneous-residual-variance-11" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-24" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;sex&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20sex%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2        se_r
1        GxE_het_var 0.7928212 0.009319443
2 GxE_by_chr_het_var 0.6872563 0.008567386</code></pre>
</div>
</div>
</div>
</div>
<div id="new-temperature-and-sex-scenario" class="section level3">
<h3>New temperature and sex scenario</h3>
<p>In this scenario, we assigned all the observations in a specific
temperature and sex to the test set, i.e., that temperature/sex
combination is never seen in the training set.</p>
<div id="standard-models-5" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance-13" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-25" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 6
schemes &lt;- &quot;temp_sex&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(temp, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 30 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 30 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sex-orig-1">
Past versions of "sex orig-1.png"
</button>
</p>
<div id="fig-no-spaces-sex-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9a93c45b800f43a4f4c6d14619221bf65971f03e/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex orig-1.png" target="_blank">9a93c45</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sex-orig-2">
Past versions of "sex orig-2.png"
</button>
</p>
<div id="fig-no-spaces-sex-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9a93c45b800f43a4f4c6d14619221bf65971f03e/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex orig-2.png" target="_blank">9a93c45</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model   mean_r2       se_r
1     E 0.0000000 0.00000000
2     G 0.3926885 0.01689340
3 GandE 0.4156498 0.01872787
4   GxE 0.4361919 0.02644908</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>   temp   scheme model        r2
1     1 temp_sex     G 0.5351017
2     2 temp_sex     G 0.4592271
3     3 temp_sex     G 0.3429214
4     4 temp_sex     G 0.4221800
5     5 temp_sex     G 0.2458393
6     6 temp_sex     G 0.3508613
7     1 temp_sex     E 0.0000000
8     2 temp_sex     E 0.0000000
9     3 temp_sex     E 0.0000000
10    4 temp_sex     E 0.0000000
11    5 temp_sex     E 0.0000000
12    6 temp_sex     E 0.0000000
13    1 temp_sex GandE 0.5533598
14    2 temp_sex GandE 0.5227960
15    3 temp_sex GandE 0.3704011
16    4 temp_sex GandE 0.4290736
17    5 temp_sex GandE 0.2468092
18    6 temp_sex GandE 0.3714590
19    1 temp_sex   GxE 0.6346551
20    2 temp_sex   GxE 0.6253384
21    3 temp_sex   GxE 0.3671544
22    4 temp_sex   GxE 0.4200270
23    5 temp_sex   GxE 0.2862420
24    6 temp_sex   GxE 0.2837347</code></pre>
</div>
<div id="phenotype-scaled-within-environment-5" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>n_reps &lt;- 6
schemes &lt;- &quot;temp_sex&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;scaled_within_env_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;temp&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(temp, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 30 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 30 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20within%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sex-within-env-1">
Past versions of "sex within env-1.png"
</button>
</p>
<div id="fig-no-spaces-sex-within-env-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9a93c45b800f43a4f4c6d14619221bf65971f03e/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex within env-1.png" target="_blank">9a93c45</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20within%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sex-within-env-2">
Past versions of "sex within env-2.png"
</button>
</p>
<div id="fig-no-spaces-sex-within-env-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/9a93c45b800f43a4f4c6d14619221bf65971f03e/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex within env-2.png" target="_blank">9a93c45</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model   mean_r2        se_r
1     E 0.0000000 0.000000000
2     G 0.4261341 0.007916038
3 GandE 0.4259122 0.007851941
4   GxE 0.4623860 0.007786653</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>   temp   scheme model        r2
1     1 temp_sex     G 0.4274009
2     2 temp_sex     G 0.4094969
3     3 temp_sex     G 0.4292001
4     4 temp_sex     G 0.4805776
5     5 temp_sex     G 0.3453162
6     6 temp_sex     G 0.4648126
7     1 temp_sex     E 0.0000000
8     2 temp_sex     E 0.0000000
9     3 temp_sex     E 0.0000000
10    4 temp_sex     E 0.0000000
11    5 temp_sex     E 0.0000000
12    6 temp_sex     E 0.0000000
13    1 temp_sex GandE 0.4274546
14    2 temp_sex GandE 0.4095090
15    3 temp_sex GandE 0.4280742
16    4 temp_sex GandE 0.4790004
17    5 temp_sex GandE 0.3457700
18    6 temp_sex GandE 0.4656650
19    1 temp_sex   GxE 0.5030803
20    2 temp_sex   GxE 0.5082190
21    3 temp_sex   GxE 0.4076840
22    4 temp_sex   GxE 0.4578788
23    5 temp_sex   GxE 0.4053029
24    6 temp_sex   GxE 0.4921512</code></pre>
<p>With the phenotype on the original scale, in general, prediction
accuracy is lower in this scenario. As expected, E-BLUP does not have
any predictive ability; all the observations in the test set get the
same predicted phenotype because they are equally “related” based on the
environment. G-BLUP predicts decently because all the lines are present
in the test set. GE-BLUP barely improves accuracy over G-BLUP. GxE-BLUP
is better than G-BLUP and GE-BLUP, on average.</p>
<p>With the phenotype scaled within environment, the pattern is very
similar in this scenario, with the improvement of using GxE being
larger.</p>
</div>
</div>
<div id="heterogeneous-residual-variance-12" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-26" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 6
schemes &lt;- &quot;temp_sex&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sex-orig-het-1">
Past versions of "sex orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-sex-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across temp/sex orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model   mean_r2        se_r
1     E_het_var 0.0000000 0.000000000
2     G_het_var 0.2736952 0.004330556
3 GandE_het_var 0.3616820 0.010050622
4   GxE_het_var 0.4088251 0.006291057</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-gwas-informed-models-4" class="section level4">
<h4>GxE GWAS-informed models</h4>
<div id="homogeneous-residual-variance-14" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-27" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 6
schemes &lt;- &quot;temp_sex&quot;
model &lt;- &#39;GxE_gwas&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>     model  pval   mean_r2       se_r
1 GxE_gwas   all 0.4361919 0.02644908
2 GxE_gwas  1e-3 0.4254371 0.02959882
3 GxE_gwas  1e-5 0.4246526 0.02987023
4 GxE_gwas  1e-8 0.4243049 0.02985203
5 GxE_gwas 1e-10 0.4269488 0.02913486
6 GxE_gwas 1e-12 0.4319157 0.02588019
7 GxE_gwas 1e-14 0.4439631 0.02414527</code></pre>
<p>…</p>
</div>
</div>
<div id="heterogeneous-residual-variance-13" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-28" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 6
schemes &lt;- &quot;temp_sex&quot;
model &lt;- &#39;GxE_gwas_het_var&#39;
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 5
n_row &lt;- n_reps * length(schemes) * length(pvals)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;pval&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv==&quot;all&quot;){
          dat &lt;- readRDS(paste0(&quot;output/GxE_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_GxE_fit_&quot;, repp, &quot;.rds&quot;))
        } else {
          dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met,&quot;_&quot;, pv, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
        }
        i &lt;- i + 1
        
        res[i, 1] &lt;- repp
        res[i, 2] &lt;- sche
        res[i, 3] &lt;- met
        res[i, 4] &lt;- pv
        if(!is.null(dat)){
          res[i, 5] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] &lt;- NA
        }
      }
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p &lt;- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20het%20gwas-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model, pval) %&gt;% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;model&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>             model  pval   mean_r2       se_r
1 GxE_gwas_het_var   all 0.4361919 0.02644908
2 GxE_gwas_het_var  1e-3 0.4014195 0.02657523
3 GxE_gwas_het_var  1e-5 0.4036921 0.02928129
4 GxE_gwas_het_var  1e-8 0.4048188 0.03061818
5 GxE_gwas_het_var 1e-10 0.4102033 0.02950911
6 GxE_gwas_het_var 1e-12 0.4103570 0.02376445
7 GxE_gwas_het_var 1e-14 0.4190064 0.01473660</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
<div id="gxe-by-chromosomes-models-4" class="section level4">
<h4>GxE by chromosomes models</h4>
<div id="homogeneous-residual-variance-15" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-29" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;temp_sex&quot;
model &lt;- c(&quot;GxE&quot;, &quot;GxE_by_chr&quot;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_1xE&quot;, &quot;VarG_2xE&quot;, &quot;VarG_3xE&quot;, 
                       &quot;VarG_4xE&quot;, &quot;VarG_5xE&quot;, &quot;VarG_6xE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] &lt;- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] &lt;- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] &lt;- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] &lt;- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] &lt;- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] &lt;- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] &lt;- dat$model_fit$varE
      res_var[i, 14] &lt;- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propG_1xE &lt;- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE &lt;- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE &lt;- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE &lt;- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE &lt;- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE &lt;- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propG_1xE&quot;, &quot;propG_2xE&quot;, &quot;propG_3xE&quot;, 
                                                &quot;propG_4xE&quot;, &quot;propG_5xE&quot;, &quot;propG_6xE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), 
                               labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarG_2LxE&quot;, &quot;VarG_2RxE&quot;, &quot;VarG_3LxE&quot;, 
                                        &quot;VarG_3RxE&quot;, &quot;VarG_XxE&quot;, &quot;VarG_4xE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  #scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 36 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 36 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20chr-2.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>       model   mean_r2       se_r
1        GxE 0.4361919 0.02644908
2 GxE_by_chr 0.4242213 0.02831674</code></pre>
</div>
</div>
<div id="heterogeneous-residual-variance-14" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-30" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>schemes &lt;- &quot;temp_sex&quot;
model &lt;- c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c(&#39;GxE_het_var&#39;, &#39;GxE_by_chr_het_var&#39;)))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20temp/sex%20orig%20het%20chr-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>               model   mean_r2        se_r
1        GxE_het_var 0.4088251 0.006291057
2 GxE_by_chr_het_var 0.3684599 0.014506093</code></pre>
</div>
</div>
</div>
</div>
<div id="gradient-1-and-2-scenario" class="section level3">
<h3>Gradient 1 and 2 scenario</h3>
<p>In this scenario, …</p>
<div id="standard-models-6" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance-16" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-31" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;gradient_1_2&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad1/2%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-2-orig-1">
Past versions of "2 orig-1.png"
</button>
</p>
<div id="fig-no-spaces-2-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad1/2 orig-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad1/2%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-2-orig-2">
Past versions of "2 orig-2.png"
</button>
</p>
<div id="fig-no-spaces-2-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad1/2 orig-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model   mean_r2       se_r
1     E 0.4877220 0.01388916
2     G 0.2079467 0.01155694
3 GandE 0.5025922 0.06661142
4   GxE 0.5304191 0.04604954</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>  rep       scheme model        r2
1   1 gradient_1_2     G 0.1916028
2   2 gradient_1_2     G 0.2242907
3   1 gradient_1_2     E 0.4680798
4   2 gradient_1_2     E 0.5073642
5   1 gradient_1_2 GandE 0.4083895
6   2 gradient_1_2 GandE 0.5967950
7   1 gradient_1_2   GxE 0.4652952
8   2 gradient_1_2   GxE 0.5955430</code></pre>
</div>
<div id="phenotype-scaled-within-environment-6" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;gradient_1_2&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;scaled_within_env_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad1/2%20within%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-2-within-env-1">
Past versions of "2 within env-1.png"
</button>
</p>
<div id="fig-no-spaces-2-within-env-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad1/2 within env-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad1/2%20within%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-2-within-env-2">
Past versions of "2 within env-2.png"
</button>
</p>
<div id="fig-no-spaces-2-within-env-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad1/2 within env-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model     mean_r2         se_r
1     E 0.001960104 0.0000985436
2     G 0.235953854 0.0115546724
3 GandE 0.237130495 0.0110332587
4   GxE 0.235450104 0.0194963869</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>  rep       scheme model          r2
1   1 gradient_1_2     G 0.219613080
2   2 gradient_1_2     G 0.252294628
3   1 gradient_1_2     E 0.001820742
4   2 gradient_1_2     E 0.002099465
5   1 gradient_1_2 GandE 0.221527111
6   2 gradient_1_2 GandE 0.252733879
7   1 gradient_1_2   GxE 0.207878050
8   2 gradient_1_2   GxE 0.263022159</code></pre>
<p>With the phenotype on the original scale, …</p>
<p>With the phenotype scaled within environment, …</p>
</div>
</div>
<div id="heterogeneous-residual-variance-15" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-32" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;gradient_1_2&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad1/2%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-2-orig-het-1">
Past versions of "2 orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-2-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad1/2 orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model   mean_r2        se_r
1     E_het_var 0.4934274 0.008841599
2     G_het_var 0.1021769 0.012736469
3 GandE_het_var 0.5237888 0.047787222
4   GxE_het_var 0.5388347 0.034080272</code></pre>
<p>With heterogeneous residual variance, …</p>
</div>
</div>
</div>
</div>
<div id="gradient-3-and-4-scenario" class="section level3">
<h3>Gradient 3 and 4 scenario</h3>
<p>In this scenario, …</p>
<div id="standard-models-7" class="section level4">
<h4>Standard models</h4>
<div id="homogeneous-residual-variance-17" class="section level5">
<h5>Homogeneous residual variance</h5>
<div id="original-scale-phenotype-33" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;gradient_3_4&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad3/4%20orig-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-4-orig-1">
Past versions of "4 orig-1.png"
</button>
</p>
<div id="fig-no-spaces-4-orig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad3/4 orig-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad3/4%20orig-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-4-orig-2">
Past versions of "4 orig-2.png"
</button>
</p>
<div id="fig-no-spaces-4-orig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad3/4 orig-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model   mean_r2         se_r
1     E 0.4192114 0.0350159552
2     G 0.1349444 0.0005952573
3 GandE 0.4878010 0.0659951952
4   GxE 0.4691867 0.0770864681</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>  rep       scheme model        r2
1   1 gradient_3_4     G 0.1357863
2   2 gradient_3_4     G 0.1341026
3   1 gradient_3_4     E 0.3696914
4   2 gradient_3_4     E 0.4687315
5   1 gradient_3_4 GandE 0.3944697
6   2 gradient_3_4 GandE 0.5811323
7   1 gradient_3_4   GxE 0.3601699
8   2 gradient_3_4   GxE 0.5782034</code></pre>
</div>
<div id="phenotype-scaled-within-environment-7" class="section level6">
<h6>Phenotype scaled within environment</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;gradient_3_4&quot;
model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;scaled_within_env_pheno&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

res_var &lt;- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] &lt;- repp
      res_var[i, 2] &lt;- sche
      res_var[i, 3] &lt;- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] &lt;- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] &lt;- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] &lt;- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] &lt;- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] &lt;- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] &lt;- NA
      }

      res_var[i, 7] &lt;- dat$model_fit$varE
      res_var[i, 8] &lt;- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(rep, model, propG, propE, propGxE, propEps) %&gt;% 
  gather(value=&quot;Prop&quot;, key=&quot;Source&quot;, propG, propE, propGxE, propEps)

res_var_long &lt;- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c(&quot;propG&quot;, &quot;propE&quot;, &quot;propGxE&quot;, &quot;propEps&quot;), labels=c(&quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarRes&quot;)))

p &lt;- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;total&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite outside the scale range
(`stat_summary()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad3/4%20within%20env-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-4-within-env-1">
Past versions of "4 within env-1.png"
</button>
</p>
<div id="fig-no-spaces-4-within-env-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad3/4 within env-1.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad3/4%20within%20env-2.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-4-within-env-2">
Past versions of "4 within env-2.png"
</button>
</p>
<div id="fig-no-spaces-4-within-env-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/63b4b0c56299ea7d6ba5cf88f417155718ec5f55/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad3/4 within env-2.png" target="_blank">63b4b0c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model      mean_r2         se_r
1     E 2.230542e-24 1.355452e-24
2     G 2.934962e-01 3.341502e-03
3 GandE 2.930503e-01 3.051084e-03
4   GxE 3.117071e-01 2.854856e-03</code></pre>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% as.data.frame()</code></pre>
<pre><code>  rep       scheme model           r2
1   1 gradient_3_4     G 2.887706e-01
2   2 gradient_3_4     G 2.982218e-01
3   1 gradient_3_4     E 3.136431e-25
4   2 gradient_3_4     E 4.147441e-24
5   1 gradient_3_4 GandE 2.887355e-01
6   2 gradient_3_4 GandE 2.973652e-01
7   1 gradient_3_4   GxE 3.157445e-01
8   2 gradient_3_4   GxE 3.076697e-01</code></pre>
<p>With the phenotype on the original scale, …</p>
<p>With the phenotype scaled within environment, …</p>
</div>
</div>
<div id="heterogeneous-residual-variance-16" class="section level5">
<h5>Heterogeneous residual variance</h5>
<div id="original-scale-phenotype-34" class="section level6">
<h6>Original scale phenotype</h6>
<pre class="r"><code>n_reps &lt;- 2
schemes &lt;- &quot;gradient_3_4&quot;
model &lt;- c(&#39;G_het_var&#39;,&#39;E_het_var&#39;,&#39;GandE_het_var&#39;,&#39;GxE_het_var&#39;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r2&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, sche, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(R)^2), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_results.Rmd/across%20grad3/4%20orig%20het-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-4-orig-het-1">
Past versions of "4 orig het-1.png"
</button>
</p>
<div id="fig-no-spaces-4-orig-het-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/86e70e8f90276892cf04e441b9537e217c723826/docs/figure/dgrp_lifespan_gxe_genomic_results.Rmd/across grad3/4 orig het-1.png" target="_blank">86e70e8</a>
</td>
<td>
fmorgante
</td>
<td>
2024-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>          model    mean_r2        se_r
1     E_het_var 0.41693974 0.037003153
2     G_het_var 0.09958552 0.007419591
3 GandE_het_var 0.54148424 0.015349582
4   GxE_het_var 0.49707333 0.046989572</code></pre>
<p>With heterogeneous residual variance, …</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.5 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/libs/gnu9/openblas/0.3.7/lib/libopenblasp-r0.3.7.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.4   tidyr_1.3.1   cowplot_1.1.3 ggplot2_3.5.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.13-1     pillar_1.9.0      compiler_4.1.2    bslib_0.8.0      
 [5] later_1.3.2       jquerylib_0.1.4   git2r_0.32.0      workflowr_1.7.0  
 [9] tools_4.1.2       digest_0.6.37     gtable_0.3.6      jsonlite_1.8.9   
[13] evaluate_1.0.1    lifecycle_1.0.4   tibble_3.2.1      pkgconfig_2.0.3  
[17] rlang_1.1.4       cli_3.6.2         rstudioapi_0.16.0 yaml_2.3.10      
[21] xfun_0.49         fastmap_1.2.0     withr_3.0.2       stringr_1.5.1    
[25] knitr_1.49        generics_0.1.3    fs_1.6.5          vctrs_0.6.5      
[29] sass_0.4.9        tidyselect_1.2.1  rprojroot_2.0.4   grid_4.1.2       
[33] glue_1.8.0        R6_2.5.1          fansi_1.0.6       rmarkdown_2.29   
[37] farver_2.1.2      purrr_1.0.2       magrittr_2.0.3    whisker_0.4.1    
[41] scales_1.3.0      promises_1.3.0    htmltools_0.5.8.1 colorspace_2.1-1 
[45] httpuv_1.6.11     labeling_0.4.3    utf8_1.2.4        stringi_1.8.4    
[49] munsell_0.5.1     cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
