<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Morgante" />

<meta name="date" content="2024-10-14" />

<title>Transcriptomic prediction of life span with different models</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp_lifespan_gxe</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/morgante_lab/dgrp_lifespan_gxe">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Transcriptomic prediction of life span with
different models</h1>
<h4 class="author">Fabio Morgante</h4>
<h4 class="date">October 14, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp_lifespan_gxe/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241004code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241004code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241004)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetree4e3e612fed582fc8a5899e97a606e92f598b4343targetblank4e3e612a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/4e3e612fed582fc8a5899e97a606e92f598b4343" target="_blank">4e3e612</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetree4e3e612fed582fc8a5899e97a606e92f598b4343targetblank4e3e612a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/4e3e612fed582fc8a5899e97a606e92f598b4343" target="_blank">4e3e612</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .snakemake/
    Ignored:    output/E_fit/
    Ignored:    output/Et_fit/
    Ignored:    output/G_fit/
    Ignored:    output/GandE_fit/
    Ignored:    output/GxE_fit/
    Ignored:    output/T_fit/
    Ignored:    output/TandE_fit/
    Ignored:    output/TxE_fit/
    Ignored:    output/test_sets/
    Ignored:    run/
    Ignored:    slurm/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/dgrp_lifespan_gxe_transcriptomic_results.Rmd</code>) and
HTML (<code>docs/dgrp_lifespan_gxe_transcriptomic_results.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/4e3e612fed582fc8a5899e97a606e92f598b4343/analysis/dgrp_lifespan_gxe_transcriptomic_results.Rmd" target="_blank">4e3e612</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Minor embellishments
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/dgrp_lifespan_gxe_transcriptomic_results.html" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/24fb66c0b6810d35bbd8eb641186407bf7d2abd0/analysis/dgrp_lifespan_gxe_transcriptomic_results.Rmd" target="_blank">24fb66c</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
<td>
Update results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>###Load libraries
library(ggplot2)
library(cowplot)
library(dplyr)

prefix &lt;- &quot;dgrp_lifespan_gxe&quot;
n_reps &lt;- 20</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The goal of this analysis is to assess the performance of the linear
mixed models including gene expression only, environment only, gene
expression + environment, gene expression + environment + gene
expression <span class="math inline">\(\times\)</span> environment
information at predicting life span in the DGRP.</p>
<p>We used data from <a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000645">Huang
et al. (2020)</a>, which has life span measurements for <span
class="math inline">\(n=176\)</span> lines at 3 temperatures (18C, 25C,
28C) for the 2 sexes (coded as 1=females and 0=males), which results in
<span class="math inline">\(q=1,056\)</span> observations. The authors
showed that there is extensive <span class="math inline">\(G \times
E\)</span> affecting life span. In this analysis, we used both
temperature as the environmental variable, since the transcriptome is
different across sexes. Transcriptomic data at baseline conditions were
obtained by <a
href="https://genome.cshlp.org/content/30/3/485.short">Everett et
al. (2020)</a>.</p>
<p>We fitted the following linear mixed models:</p>
<ul>
<li>T-BLUP – <span class="math inline">\(y = \mu + t +
\epsilon\)</span>.</li>
<li>E-BLUP – <span class="math inline">\(y = \mu + e +
\epsilon\)</span>.</li>
<li>TE-BLUP – <span class="math inline">\(y = \mu + t + e +
\epsilon\)</span>.</li>
<li>TxE-BLUP – <span class="math inline">\(y = \mu + t + e + te +
\epsilon\)</span>.</li>
</ul>
<p>where</p>
<p><span class="math inline">\(y\)</span> is a <span
class="math inline">\(q\)</span>-vector of phenotypic observations,
<span class="math inline">\(\mu\)</span> is a <span
class="math inline">\(q\)</span>-vector of intercept values, <span
class="math inline">\(t\)</span> is a <span
class="math inline">\(q\)</span>-vector of transcriptomic values, <span
class="math inline">\(t \sim N_q(\mathbf O, \mathbf{ZTZ^\intercal}
\sigma^2_t)\)</span>, <span class="math inline">\(\mathbf Z\)</span> is
a <span class="math inline">\(q \times n\)</span> incidence matrix,
<span class="math inline">\(\mathbf T\)</span> is a <span
class="math inline">\(n \times n\)</span> transcriptomic relationship
matrix (TRM), <span class="math inline">\(e\)</span> is a <span
class="math inline">\(q\)</span>-vector of environmental values, <span
class="math inline">\(e \sim N_q(\mathbf O, \mathbf{E}
\sigma^2_e)\)</span>, <span class="math inline">\(\mathbf E\)</span> is
a <span class="math inline">\(q \times q\)</span> matrix of similarity
based on environmental variables, computed as <span
class="math inline">\(\mathbf E = \mathbf{XX}^\intercal\)</span>, <span
class="math inline">\(\mathbf X\)</span> is a <span
class="math inline">\(q \times c\)</span> matrix of environmental
measurements, <span class="math inline">\(te \sim N_q(\mathbf O,
\mathbf{ZTZ^\intercal \circ E} \sigma^2_{te})\)</span>, <span
class="math inline">\(\epsilon\)</span> is a <span
class="math inline">\(q\)</span>-vector of residual values, <span
class="math inline">\(\epsilon \sim N_q(\mathbf O, \mathbf{I}
\sigma^2_\epsilon)\)</span>.</p>
<p>These models were fitted in the training set using a Bayesian
approach as implemented in the <span class="math inline">\(\sf
BGLR\)</span> package.</p>
<p>We used different cross-validation schemes:</p>
<ul>
<li><em>Random within environment scenario</em>. We assigned 20% of the
<em>lines</em> to the test set randomly and analyzed each
sex/temperature combination separately using the standard GBLUP model.
This procedure was repeated 20 times.</li>
<li><em>Random lines scenario</em>. We assigned 20% of the
<em>lines</em> (i.e., for the 2 sexes and the 3 temperatures) to the
test set randomly, and analyzed the sex/temperature combinations jointly
with the models above. The peculiarity of this scenario is that the
lines in the test set are not represented at all in the training set.
This procedure was repeated 20 times.</li>
<li><em>Random observations scenario</em>. We assigned 20% of the
<em>observations</em> (i.e., combination of line, sex and temperature)
to the test set randomly, and analyzed the sex/temperature combinations
jointly with the models above. The peculiarity of this scenario is that
all the lines, sexes and temperatures are represented in the training
set. This procedure was repeated 20 times.</li>
</ul>
<p>Prediction accuracy was evaluated as the correlation coefficient
between observed and predicted phenotypes in the test set. The two sexes
were analyzed separately.</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="females" class="section level3">
<h3>Females</h3>
<div id="random-within-environment-scenario" class="section level4">
<h4>Random within environment scenario</h4>
<p>This is the baseline scenario, where we acknowldege that the genetic
architecture of life span is different across environments and we
analyze them separately using T-BLUP.</p>
<pre class="r"><code>sex &lt;- 1
temp &lt;- c(18,25,28)
model &lt;- &quot;T&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(temp) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;r&quot;)

  for(t in temp){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/T_fit/&quot;, prefix, &quot;_random_within_env_&quot;, sex, &quot;_&quot;, t, &quot;_T_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
      
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- t
      res[i, 3] &lt;- model
      res[i, 4] &lt;- cor(dat$y_test, dat$yhat_test)
    }
  }

res &lt;- transform(res, temp=as.factor(temp))

p &lt;- ggplot(res, aes(x = temp, y = r)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(r)), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random%20within%20env%20females-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-females-1">
Past versions of "random within env females-1.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-females-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random within env females-1.png" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(temp) %&gt;% summarise(mean_r=mean(r), se_r=(sd(r)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  temp    mean_r        se_r
1   18 0.3202463 0.006705572
2   25 0.2739777 0.009025876
3   28 0.1603950 0.006685838</code></pre>
<p>The results show that prediction accuracy varies across environments
and is higher than using <a
href="dgrp_lifespan_gxe_genomic_results.html">genotypes</a>.</p>
</div>
<div id="random-lines-scenario" class="section level4">
<h4>Random lines scenario</h4>
<p>In this scenario, we combine the data across environments and fit the
different models.</p>
<pre class="r"><code>sex &lt;- 1
schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&quot;T&quot;, &quot;Et&quot;, &quot;TandE&quot;, &quot;TxE&quot;)


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, sche, &quot;_&quot;, sex, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- cor(dat$y_test, dat$yhat_test)
    }
  }
}

res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  scale_x_discrete(labels=c(&quot;Et&quot; = &quot;E&quot;, &quot;T&quot; = &quot;T&quot;, &quot;TandE&quot; = &quot;TandE&quot;, &quot;TxE&quot; = &quot;TxE&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(r)), title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random%20lines%20females-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-females-1">
Past versions of "random lines females-1.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-females-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random lines females-1.png" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r=mean(r), se_r=(sd(r)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model    mean_r        se_r
1    Et 0.8484243 0.001252802
2     T 0.1176381 0.002640235
3 TandE 0.8578200 0.001216803
4   TxE 0.8643102 0.001207871</code></pre>
<p>The results show that T-BLUP performs poorly. This is expected
because we are making the implicit assumption that transcriptomic
effects are equal across environments, which is not the case. E-BLUP
does very well in this scenario because when combining the data across
environments, a lot of the variance is explained by the environmental
effects, which we have good power to estimate. TxE-BLUP gives marginally
higher accuracy than TE-BLUP, which gives marginally higher accuracy
than E-BLUP, although these accuracies are very similar.</p>
</div>
<div id="random-observations-scenario" class="section level4">
<h4>Random observations scenario</h4>
<p>In this scenario, every line and environment is potentially
represented in the training set. It is specific combinations of lines
and environments (e.g., line_101 at 18C) that are not seen in the
training set. This is the scenario where we expect an advantage from
using models with both gene expression and environment information
because the training-test transfer of information happens at every
level.</p>
<pre class="r"><code>sex &lt;- 1
schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&quot;T&quot;, &quot;Et&quot;, &quot;TandE&quot;, &quot;TxE&quot;)


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, sche, &quot;_&quot;, sex, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- cor(dat$y_test, dat$yhat_test)
    }
  }
}

res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  scale_x_discrete(labels=c(&quot;Et&quot; = &quot;E&quot;, &quot;T&quot; = &quot;T&quot;, &quot;TandE&quot; = &quot;TandE&quot;, &quot;TxE&quot; = &quot;TxE&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(r)), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random%20obs%20females-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-females-1">
Past versions of "random obs females-1.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-females-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random obs females-1.png" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r=mean(r), se_r=(sd(r)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model     mean_r        se_r
1    Et  0.8450099 0.001429850
2     T -0.2055787 0.003109942
3 TandE  0.8698321 0.001347471
4   TxE  0.8767333 0.001353551</code></pre>
<p>The results show a similar pattern to the previous scenario. However,
TE-BLUP and TxE-BLUP achieve marginally higher accuracies than in the
previous scenario, due to the full training-test transfer of
information.</p>
</div>
</div>
<div id="males" class="section level3">
<h3>Males</h3>
<div id="random-within-environment-scenario-1" class="section level4">
<h4>Random within environment scenario</h4>
<p>This is the baseline scenario, where we acknowldege that the genetic
architecture of life span is different across environments and we
analyze them separately using T-BLUP.</p>
<pre class="r"><code>sex &lt;- 0
temp &lt;- c(18,25,28)
model &lt;- &quot;T&quot;


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(temp) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;temp&quot;, &quot;model&quot;, &quot;r&quot;)

  for(t in temp){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/T_fit/&quot;, prefix, &quot;_random_within_env_&quot;, sex, &quot;_&quot;, t, &quot;_T_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
      
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- t
      res[i, 3] &lt;- model
      res[i, 4] &lt;- cor(dat$y_test, dat$yhat_test)
    }
  }

res &lt;- transform(res, temp=as.factor(temp))

p &lt;- ggplot(res, aes(x = temp, y = r)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  labs(x = &quot;Environment&quot;, y = expression(italic(r)), title=&quot;&quot;) +
  theme_cowplot(font_size = 18)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random%20within%20env%20males-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-within-env-males-1">
Past versions of "random within env males-1.png"
</button>
</p>
<div id="fig-no-spaces-random-within-env-males-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random within env males-1.png" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(temp) %&gt;% summarise(mean_r=mean(r), se_r=(sd(r)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  temp    mean_r        se_r
1   18 0.2711625 0.004284594
2   25 0.2768752 0.007115864
3   28 0.2380711 0.005792715</code></pre>
<p>The results show that prediction accuracy is pretty similar across
environments.</p>
</div>
<div id="random-lines-scenario-1" class="section level4">
<h4>Random lines scenario</h4>
<p>In this scenario, we combine the data across environments and fit the
different models.</p>
<pre class="r"><code>sex &lt;- 0
schemes &lt;- &quot;random_lines&quot;
model &lt;- c(&quot;T&quot;, &quot;Et&quot;, &quot;TandE&quot;, &quot;TxE&quot;)


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, sche, &quot;_&quot;, sex, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- cor(dat$y_test, dat$yhat_test)
    }
  }
}

res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  scale_x_discrete(labels=c(&quot;Et&quot; = &quot;E&quot;, &quot;T&quot; = &quot;T&quot;, &quot;TandE&quot; = &quot;TandE&quot;, &quot;TxE&quot; = &quot;TxE&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(r)), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random%20lines%20males-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-lines-males-1">
Past versions of "random lines males-1.png"
</button>
</p>
<div id="fig-no-spaces-random-lines-males-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random lines males-1.png" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r=mean(r), se_r=(sd(r)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model    mean_r         se_r
1    Et 0.8593359 0.0007915789
2     T 0.1025497 0.0013584120
3 TandE 0.8673822 0.0007701420
4   TxE 0.8696027 0.0008268993</code></pre>
<p>The results show that T-BLUP performs poorly. This is expected
because we are making the implicit assumption that transcriptomic
effects are equal across environments, which is not the case. E-BLUP
does very well in this scenario because when combining the data across
environments, a lot of the variance is explained by the environmental
effects, which we have good power to estimate. TE-BLUP and TxE-BLUP give
marginally higher accuracy than E-BLUP, although these accuracies are
very similar.</p>
</div>
<div id="random-observations-scenario-1" class="section level4">
<h4>Random observations scenario</h4>
<p>In this scenario, every line and environment is potentially
represented in the training set. It is specific combinations of lines
and environments (e.g., line_101 in females at 18C) that are not seen in
the training set. This is the scenario where we expect an advantage from
using models with both gene expression and environment information
because the training-test transfer of information happens at every
level</p>
<pre class="r"><code>sex &lt;- 0
schemes &lt;- &quot;random_obs&quot;
model &lt;- c(&quot;T&quot;, &quot;Et&quot;, &quot;TandE&quot;, &quot;TxE&quot;)


i &lt;- 0

n_col &lt;- 4
n_row &lt;- n_reps * length(schemes) * length(model)
res &lt;- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) &lt;- c(&quot;rep&quot;, &quot;scheme&quot;, &quot;model&quot;, &quot;r&quot;)

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, sche, &quot;_&quot;, sex, &quot;_&quot;, met, &quot;_fit_&quot;, repp, &quot;.rds&quot;))
      i &lt;- i + 1
        
      res[i, 1] &lt;- repp
      res[i, 2] &lt;- sche
      res[i, 3] &lt;- met
      res[i, 4] &lt;- cor(dat$y_test, dat$yhat_test)
    }
  }
}

res &lt;- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p &lt;- ggplot(res, aes(x = model, y = r, fill = model)) +
  geom_boxplot(color = &quot;black&quot;, outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom=&quot;point&quot;, shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = &quot;single&quot;)) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;orange&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;lightblue&quot;)) +
  scale_x_discrete(labels=c(&quot;Et&quot; = &quot;E&quot;, &quot;T&quot; = &quot;T&quot;, &quot;TandE&quot; = &quot;TandE&quot;, &quot;TxE&quot; = &quot;TxE&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(r)), fill=&quot;Method&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) +
  theme(legend.position=&quot;none&quot;)

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random%20obs%20males-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-random-obs-males-1">
Past versions of "random obs males-1.png"
</button>
</p>
<div id="fig-no-spaces-random-obs-males-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/022638e00dda28a2ce07d72485b8cd4cc32b5929/docs/figure/dgrp_lifespan_gxe_transcriptomic_results.Rmd/random obs males-1.png" target="_blank">022638e</a>
</td>
<td>
fmorgante
</td>
<td>
2024-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% group_by(model) %&gt;% summarise(mean_r=mean(r), se_r=(sd(r)/n_reps)) %&gt;% as.data.frame()</code></pre>
<pre><code>  model     mean_r         se_r
1    Et  0.8660534 0.0006771890
2     T -0.2649558 0.0030027565
3 TandE  0.8917200 0.0006220257
4   TxE  0.8922233 0.0006674030</code></pre>
<p>The results show a similar pattern to the previous scenario. However,
the difference between (TE-BLUP, TxE-BLUP) and E-BLUP is larger than in
the previous scenario, due to the full training-test transfer of
information.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.5 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/libs/gnu9/openblas/0.3.7/lib/libopenblasp-r0.3.7.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.4   cowplot_1.1.3 ggplot2_3.5.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.13       highr_0.11        pillar_1.9.0      compiler_4.1.2   
 [5] bslib_0.7.0       later_1.3.2       jquerylib_0.1.4   git2r_0.32.0     
 [9] workflowr_1.7.0   tools_4.1.2       digest_0.6.35     gtable_0.3.5     
[13] jsonlite_1.8.8    evaluate_0.23     lifecycle_1.0.4   tibble_3.2.1     
[17] pkgconfig_2.0.3   rlang_1.1.4       cli_3.6.2         rstudioapi_0.16.0
[21] yaml_2.3.8        xfun_0.44         fastmap_1.2.0     withr_3.0.0      
[25] stringr_1.5.1     knitr_1.47        generics_0.1.3    fs_1.6.4         
[29] vctrs_0.6.5       sass_0.4.9        tidyselect_1.2.1  rprojroot_2.0.4  
[33] grid_4.1.2        glue_1.7.0        R6_2.5.1          fansi_1.0.6      
[37] rmarkdown_2.27    farver_2.1.2      magrittr_2.0.3    whisker_0.4.1    
[41] scales_1.3.0      promises_1.3.0    htmltools_0.5.8.1 colorspace_2.1-0 
[45] httpuv_1.6.11     labeling_0.4.3    utf8_1.2.4        stringi_1.8.4    
[49] munsell_0.5.1     cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
