<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Morgante" />

<meta name="date" content="2025-04-16" />

<title>Variance partition of life span with different models</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp_lifespan_gxe</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/morgante_lab/dgrp_lifespan_gxe">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Variance partition of life span with
different models</h1>
<h4 class="author">Fabio Morgante</h4>
<h4 class="date">April 16, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp_lifespan_gxe/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241004code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241004code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241004)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetree1c45a58ffca752ec1a1ef472e947b1fc0a43f86dtargetblank1c45a58a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/1c45a58ffca752ec1a1ef472e947b1fc0a43f86d" target="_blank">1c45a58</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommorgantelabdgrplifespangxetree1c45a58ffca752ec1a1ef472e947b1fc0a43f86dtargetblank1c45a58a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/tree/1c45a58ffca752ec1a1ef472e947b1fc0a43f86d" target="_blank">1c45a58</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .snakemake/
    Ignored:    output/E_fit/
    Ignored:    output/E_het_var_fit/
    Ignored:    output/G_fit/
    Ignored:    output/G_het_var_fit/
    Ignored:    output/GandE_fit/
    Ignored:    output/GandE_het_var_fit/
    Ignored:    output/GxE_by_chr_fit/
    Ignored:    output/GxE_by_chr_het_var_fit/
    Ignored:    output/GxE_fit/
    Ignored:    output/GxE_gwas/
    Ignored:    output/GxE_gwas_fit/
    Ignored:    output/GxE_gwas_het_var_fit/
    Ignored:    output/GxE_het_var_fit/
    Ignored:    output/mvgblup_fit/
    Ignored:    output/rrm_fit/
    Ignored:    output/test_sets/
    Ignored:    run/
    Ignored:    slurm/

Untracked files:
    Untracked:  analysis/dgrp_lifespan_gxe_genomic_results_OLD.Rmd
    Untracked:  scripts/run_fit_gwas_gxe_grammar_indicator.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_means.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_sex.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_grammar_temp.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lm.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_2envs_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_indicator.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_indicator_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_means.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_means_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_scaled_within_env_pheno_2envs_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_sex.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_sex_stand.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_temp.sbatch
    Untracked:  scripts/run_fit_gwas_gxe_lmm_temp_stand.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd</code>)
and HTML
(<code>docs/dgrp_lifespan_gxe_genomic_varcomps_results.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/1c45a58ffca752ec1a1ef472e947b1fc0a43f86d/analysis/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd" target="_blank">1c45a58</a>
</td>
<td>
fmorgante
</td>
<td>
2025-04-16
</td>
<td>
Small fix
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/523df1f5ac28cd946dc70afcfe5ace8db762318f/docs/dgrp_lifespan_gxe_genomic_varcomps_results.html" target="_blank">523df1f</a>
</td>
<td>
fmorgante
</td>
<td>
2025-04-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/e0a570b4e4a54d8ca4768bb69fd6027fb960b59f/analysis/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd" target="_blank">e0a570b</a>
</td>
<td>
fmorgante
</td>
<td>
2025-04-16
</td>
<td>
Update results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/morgantelab/dgrp_lifespan_gxe/ec63d8c36ff9b244df871f1f22679d24510d5df1/docs/dgrp_lifespan_gxe_genomic_varcomps_results.html" target="_blank">ec63d8c</a>
</td>
<td>
fmorgante
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/a860e4ff06b84e08caf935438a40ad810ff9fefa/analysis/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd" target="_blank">a860e4f</a>
</td>
<td>
fmorgante
</td>
<td>
2025-01-30
</td>
<td>
Add variance partition
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>###Load libraries
library(ggplot2)
library(cowplot)
library(tidyr)
library(dplyr)

prefix &lt;- &quot;dgrp_lifespan_gxe&quot;</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The goal of this analysis is to partition the phenotypic variance
using linear mixed models including genotype only, environment only,
genotype + environment, genotype + environment + genotype <span
class="math inline">\(\times\)</span> environment information.</p>
<p>We used data from <a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000645">Huang
et al. (2020)</a>, which has life span measurements for <span
class="math inline">\(n=176\)</span> lines at 3 temperatures (18C, 25C,
28C) for the 2 sexes (coded as 1=females and 0=males), which results in
<span class="math inline">\(q=1,056\)</span> observations. The authors
showed that there is extensive <span class="math inline">\(G \times
E\)</span> affecting life span. In the present work, we used both
temperature and sex as <span class="math inline">\(c=2\)</span>
environmental variables, for a total of <span
class="math inline">\(r=6\)</span> environments.</p>
<p>We fitted the following linear mixed models:</p>
<ul>
<li>G-BLUP – <span class="math inline">\(y = \mu + g +
\epsilon\)</span>.</li>
<li>E-BLUP – <span class="math inline">\(y = \mu + e +
\epsilon\)</span>.</li>
<li>GE-BLUP – <span class="math inline">\(y = \mu + g + e +
\epsilon\)</span>.</li>
<li>GxE-BLUP – <span class="math inline">\(y = \mu + g + e + ge +
\epsilon\)</span>.</li>
</ul>
<p>where</p>
<p><span class="math inline">\(y\)</span> is a <span
class="math inline">\(q\)</span>-vector of phenotypic observations,
<span class="math inline">\(\mu\)</span> is a <span
class="math inline">\(q\)</span>-vector of intercept values, <span
class="math inline">\(g\)</span> is a <span
class="math inline">\(q\)</span>-vector of additive genetic values,
<span class="math inline">\(g \sim N_q(\mathbf O, \mathbf{ZGZ^\intercal}
\sigma^2_g)\)</span>, <span class="math inline">\(\mathbf Z\)</span> is
a <span class="math inline">\(q \times n\)</span> incidence matrix,
<span class="math inline">\(\mathbf G\)</span> is a <span
class="math inline">\(n \times n\)</span> genomic relationship matrix
(GRM), <span class="math inline">\(e\)</span> is a <span
class="math inline">\(q\)</span>-vector of environmental values, <span
class="math inline">\(e \sim N_q(\mathbf O, \mathbf{E}
\sigma^2_e)\)</span>, <span class="math inline">\(\mathbf E\)</span> is
a <span class="math inline">\(q \times q\)</span> matrix of similarity
based on environmental variables, computed as <span
class="math inline">\(\mathbf E = \mathbf{XX}^\intercal\)</span>, <span
class="math inline">\(\mathbf X\)</span> is a <span
class="math inline">\(q \times c\)</span> matrix of environmental
measurements, <span class="math inline">\(ge \sim N_q(\mathbf O,
\mathbf{ZGZ^\intercal \circ E} \sigma^2_{ge})\)</span>, <span
class="math inline">\(\epsilon\)</span> is a <span
class="math inline">\(q\)</span>-vector of residual values, <span
class="math inline">\(\epsilon \sim N_q(\mathbf O, \mathbf{I}
\sigma^2_\epsilon)\)</span>.</p>
<p>These models were fitted to the whole data using a Bayesian approach
as implemented in the <span class="math inline">\(\sf BGLR\)</span>
package.</p>
<p>We also computed genetic correlations across the environments using
multivariate GBLUP – <span class="math inline">\(Y = M + U +
R\)</span>:</p>
<p>where</p>
<p><span class="math inline">\(Y\)</span> is an <span
class="math inline">\(n \times r\)</span> matrix of phenotypic
observations, <span class="math inline">\(M\)</span> is an <span
class="math inline">\(r\)</span>-vector of intercept values, <span
class="math inline">\(U\)</span> is an <span class="math inline">\(n
\times r\)</span> matrix of additive genetic values, <span
class="math inline">\(U \sim MN_{n \times r} (\mathbf O, \mathbf G,
\mathbf \Sigma_U)\)</span> and <span class="math inline">\(\mathbf
\Sigma_U\)</span> is an <span class="math inline">\(r \times r\)</span>
covariance matrix, <span class="math inline">\(R\)</span> is an <span
class="math inline">\(r \times r\)</span> matrix of residual values,
<span class="math inline">\(R \sim MN_{n \times r} (\mathbf O, \mathbf
I_n, \mathbf \Sigma_R)\)</span> and <span class="math inline">\(\mathbf
\Sigma_R\)</span> is an <span class="math inline">\(r \times r\)</span>
covariance matrix.</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<pre class="r"><code>model &lt;- c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)
pheno &lt;- &quot;original_pheno&quot;

i &lt;- 0

res_var &lt;- as.data.frame(matrix(NA, ncol=6, nrow=4))
colnames(res_var) &lt;- c(&quot;model&quot;, &quot;VarG&quot;, &quot;VarE&quot;, &quot;VarGxE&quot;, &quot;VarEps&quot;, &quot;VarTot&quot;)


for(met in model){
  dat &lt;- readRDS(paste0(&quot;output/&quot;, met, &quot;_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_&quot;, met, &quot;_fit_whole_data.rds&quot;))
  i &lt;- i + 1

  res_var[i, 1] &lt;- met
  if(!is.null(dat$ETA$G$varU)){
    res_var[i, 2] &lt;- dat$ETA$G$varU
  } else {
    res_var[i, 2] &lt;- NA
  }
      
  if(!is.null(dat$ETA$E$varU)){
    res_var[i, 3] &lt;- dat$ETA$E$varU
  } else {
    res_var[i, 3] &lt;- NA
  }

  if(!is.null(dat$ETA$GxE$varU)){
    res_var[i, 4] &lt;- dat$ETA$GxE$varU
  } else {
    res_var[i, 4] &lt;- NA
  }

  res_var[i, 5] &lt;- dat$varE
  res_var[i, 6] &lt;- sum(res_var[i, 2:5], na.rm=TRUE)
}


###PVE
res_var$propG &lt;- res_var$VarG/res_var$VarTot
res_var$propE &lt;- res_var$VarE/res_var$VarTot
res_var$propGxE &lt;- res_var$VarGxE/res_var$VarTot
res_var$propEps &lt;- res_var$VarEps/res_var$VarTot

res_var_long &lt;- res_var %&gt;% select(model, VarG, VarE, VarGxE, VarEps) %&gt;% 
  gather(value=&quot;Var&quot;, key=&quot;Source&quot;, VarG, VarE, VarGxE, VarEps)

res_var_long &lt;- transform(res_var_long, model=factor(model, levels=c(&quot;G&quot;, &quot;E&quot;, &quot;GandE&quot;, &quot;GxE&quot;)),
                 Source=factor(Source, levels=c(&quot;VarEps&quot;, &quot;VarGxE&quot;, &quot;VarE&quot;, &quot;VarG&quot;), 
                               labels=c(&quot;VarRes&quot;, &quot;VarGxE&quot;, &quot;VarE&quot;, &quot;VarG&quot;)))



p &lt;- ggplot(res_var_long, aes(x = model, y = Var, fill = Source)) +
  geom_bar(position=&quot;fill&quot;, stat=&quot;identity&quot;) +
  scale_fill_manual(values = c(&quot;lightgreen&quot;, &quot;orange&quot;, &quot;pink&quot;, &quot;blue&quot;)) +
  labs(x = &quot;Model&quot;, y = expression(italic(PVE)), fill=&quot;Source&quot;, title=&quot;&quot;) +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<pre><code>Warning: Removed 5 rows containing missing values or values outside the scale range
(`geom_bar()`).</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd/varcomps-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-varcomps-1">
Past versions of varcomps-1.png
</button>
</p>
<div id="fig-varcomps-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/ec63d8c36ff9b244df871f1f22679d24510d5df1/docs/figure/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd/varcomps-1.png" target="_blank">ec63d8c</a>
</td>
<td>
fmorgante
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There is definitely evidence of GxE.</p>
<pre class="r"><code>pheno &lt;- &quot;original_pheno&quot;

dat &lt;- readRDS(paste0(&quot;output/mvgblup_fit/&quot;, prefix, &quot;_&quot;, pheno, &quot;_mvgblup_fit_whole_data.rds&quot;))

corg &lt;- cov2cor(dat$ETA$G$Cov$Omega)
colnames(corg) &lt;- rownames(corg) &lt;- gsub(&quot;y_&quot;, &quot;&quot;, colnames(dat$ETAHat))

corg_melt &lt;-reshape2::melt(corg)
colnames(corg_melt) &lt;- c(&quot;Env1&quot;, &quot;Env2&quot;, &quot;Cor&quot;)

p &lt;- ggplot(corg_melt, aes(x = Env1, y = Env2, fill = Cor)) +
  geom_tile() +
  labs(x = &quot;&quot;, y = &quot;&quot;, fill=&quot;Cor&quot;, title=&quot;&quot;) +
  scale_fill_gradient2(low = &quot;#075AFF&quot;,
                       mid = &quot;#FFFFCC&quot;,
                       high = &quot;#FF0000&quot;) +
  coord_fixed() +
  theme_cowplot(font_size = 18) 

print(p)</code></pre>
<p><img src="figure/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd/gencor-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gencor-1">
Past versions of gencor-1.png
</button>
</p>
<div id="fig-gencor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/morgantelab/dgrp_lifespan_gxe/blob/523df1f5ac28cd946dc70afcfe5ace8db762318f/docs/figure/dgrp_lifespan_gxe_genomic_varcomps_results.Rmd/gencor-1.png" target="_blank">523df1f</a>
</td>
<td>
fmorgante
</td>
<td>
2025-04-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This is also confirmed by the estimates of the cross-environment
genetic correlations, which are different from unity in the majority of
comparisons.</p>
<p>The residual variances are also different across environments
92.9131651, 21.4166144, 9.7341183, 102.8613531, 30.4585922, 10.0329902,
justifying fitting models with heterogenous residual variance.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.5 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/libs/gnu9/openblas/0.3.7/lib/libopenblasp-r0.3.7.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.4   tidyr_1.3.1   cowplot_1.1.3 ggplot2_3.5.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.13-1     plyr_1.8.9        pillar_1.9.0      compiler_4.1.2   
 [5] bslib_0.8.0       later_1.3.2       jquerylib_0.1.4   git2r_0.32.0     
 [9] workflowr_1.7.0   tools_4.1.2       digest_0.6.37     gtable_0.3.6     
[13] jsonlite_1.8.9    evaluate_1.0.1    lifecycle_1.0.4   tibble_3.2.1     
[17] pkgconfig_2.0.3   rlang_1.1.4       cli_3.6.2         rstudioapi_0.17.1
[21] yaml_2.3.10       xfun_0.49         fastmap_1.2.0     withr_3.0.2      
[25] stringr_1.5.1     knitr_1.49        generics_0.1.3    fs_1.6.5         
[29] vctrs_0.6.5       sass_0.4.9        tidyselect_1.2.1  rprojroot_2.0.4  
[33] grid_4.1.2        glue_1.8.0        R6_2.6.1          fansi_1.0.6      
[37] rmarkdown_2.29    reshape2_1.4.4    farver_2.1.2      purrr_1.0.2      
[41] magrittr_2.0.3    whisker_0.4.1     scales_1.3.0      promises_1.3.0   
[45] htmltools_0.5.8.1 colorspace_2.1-1  httpuv_1.6.11     labeling_0.4.3   
[49] utf8_1.2.4        stringi_1.8.4     munsell_0.5.1     cachem_1.1.0     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
