---
title: "Genomic prediction of life span with different models"
author: "Fabio Morgante"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: 
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

```{r set options, message=FALSE}
###Load libraries
library(ggplot2)
library(cowplot)
library(tidyr)
library(dplyr)

prefix <- "dgrp_lifespan_gxe"
n_reps <- 20
pvals <- c("all", "1e-3", "1e-5", "1e-8", "1e-10", "1e-12", "1e-14")
```

## Introduction

The goal of this analysis is to assess the performance of the linear mixed models including genotype only, environment only,
genotype + environment, genotype + environment + genotype $\times$ environment information at predicting life span in the 
DGRP.

We used data from [Huang et al. (2020)](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000645), which
has life span measurements for $n=176$ lines at 3 temperatures (18C, 25C, 28C) for the 2 sexes (coded as 1=females and 0=males), 
which results in $q=1,056$ observations. 
The authors showed that there is extensive $G \times E$ affecting life span. In the present work, we used both temperature 
and sex as $c=2$ environmental variables, for a total of $r=6$ environments.

We fitted the following linear mixed models:

* G-BLUP -- $y = \mu + g + \epsilon$.
* E-BLUP -- $y = \mu + e + \epsilon$.
* GE-BLUP -- $y = \mu + g + e + \epsilon$.
* GxE-BLUP -- $y = \mu + g + e + ge + \epsilon$.

where

$y$ is a $q$-vector of phenotypic observations, 
$\mu$ is a $q$-vector of intercept values, 
$g$ is a $q$-vector of additive genetic values, $g \sim N_q(\mathbf O, \mathbf{ZGZ^\intercal} \sigma^2_g)$, 
$\mathbf Z$ is a $q \times n$ incidence matrix, 
$\mathbf G$ is a $n \times n$ genomic relationship matrix (GRM), 
$e$ is a $q$-vector of environmental values, $e \sim N_q(\mathbf O, \mathbf{E} \sigma^2_e)$, 
$\mathbf E$ is a $q \times q$  matrix of similarity based on environmental variables, computed as 
$\mathbf E = \mathbf{XX}^\intercal$, 
$\mathbf X$ is a $q \times c$ matrix of environmental measurements,
$ge \sim N_q(\mathbf O, \mathbf{ZGZ^\intercal \circ E} \sigma^2_{ge})$,
$\epsilon$ is a $q$-vector of residual values, $\epsilon \sim N_q(\mathbf O, \mathbf{I} \sigma^2_\epsilon)$.

These models were also implemented with heterogeneous residual variances across environments (i.e., $\epsilon \sim N_q(\mathbf O, \mathbf{D})$, where $D$ is a $r \times r$ diagonal matrix where $D_{s,s} = \sigma^2_{\epsilon s}$).

We also fit a multivariate GBLUP -- $Y = M + U + R$:

where

$Y$ is an $n \times r$ matrix of phenotypic observations, 
$M$ is an $r$-vector of intercept values, 
$U$ is an $n \times r$ matrix of additive genetic values, $U \sim MN_{n \times r}(\mathbf O, \mathbf G, \mathbf \Sigma_U})$ and $\Sigma_U$ is an $r \times r$ covariance matrix,
$R$ is an $r \times r$ matrix of residual values, $R \sim MN_{n \times r}(\mathbf O, \mathbf I_n, \mathbf \Sigma_R})$ and $\Sigma_R$ is an $r \times r$ covariance matrix.

We also fit a random regression model -- $y_{is} = \sum_{m=0}^K \beta_{m} \phi_m(s) + \sum_{m=0}^K a_{im} \phi_m(s) + \epsilon$:

where

$y_{is}$ is the phenotype for line $i$ in environment $s$, 
$\phi_m(s)$ is the Legendre polynomial of order $m$ for environment $s$,
$\beta_{m}$ is the fixed effect of the $m^{th}$-order Legendre polynomial,
$a_{im}$ is the random additive genetic value of the $m^{th}$-order Legendre polynomial, $a_{im} \sim N(\mathbf 0, \mathbf G \otimes \mathbf \Sigma_a)$ and $\Sigma_r$ is an $K \times K$ covariance matrix.
$\epsilon_{is}$ is the residual value for line $i$ in environment $s$, $\epsilon \sim N(O, \sigma^2_{\epsilon s})$.

After some testing, we chose $K=1$.

These models were fitted in the training set using a Bayesian approach as implemented in the $\sf BGLR$ package.

We used different cross-validation schemes:

* _Random within environment scenario_. We assigned 20\% of the *lines* to the test set randomly and analyzed each sex/temperature 
combination separately using the standard GBLUP model. This procedure was repeated 20 times.
* _Random lines scenario_. We assigned 20\% of the *lines* (i.e., for the 2 sexes and the 3 temperatures) to the test set randomly, 
and analyzed the sex/temperature combinations jointly with the models above. The peculiarity of this scenario is that the lines in 
the test set are not represented at all in the training set. This procedure was repeated 20 times.
* _Random observations scenario_. We assigned 20\% of the *observations* (i.e., combination of line, sex and temperature) to 
the test set randomly, and analyzed the sex/temperature combinations jointly with the models above. The peculiarity of this 
scenario is that all the lines, sexes and temperatures are represented in the training set. This procedure was repeated 20 times.
* _New temperature scenario_. We assigned all the observations in a specific temperature to the test set, i.e., that temperature 
is never seen in the training set. This procedure was repeated 3 times, once for each temperature.
* _New temperature and sex scenario_. We assigned all the observations in a specific temperature and sex to the test set, i.e., that combination 
is never seen in the training set. This procedure was repeated 6 times, once for each temperature/sex combination.

We used different treatments of the phenotype:

* Leave it on the original scale.
* Scale it within each sex/temperature combination.

Prediction accuracy was evaluated as the $R^2$ of the regression of observed phenotypes on predicted phenotypes in the test set.

## Results

### Random within environment scenario

This is the baseline scenario, where we acknowldege that the genetic architecture of life span is different across environments and we
analyze them separately using G-BLUP.

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r random within env orig, fig.height=10, fig.width=13}
sex <- c(1, 0)
temp <- c(18,25,28)
model <- "G"
pheno <- "original_pheno"


i <- 0

n_col <- 6
n_row <- n_reps * length(sex) * length(temp) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "sex", "temp", "model", "r2", "env")

res_var <- as.data.frame(matrix(NA, ncol=n_col+2, nrow=n_row))
colnames(res_var) <- c("rep", "sex", "temp", "model", "VarG", "VarEps", "H2", "env")


for(s in sex){
  for(t in temp){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/G_fit/", prefix, "_", pheno, "_random_within_env_", s, "_", t, "_G_fit_", repp, ".rds"))
      i <- i + 1
      
      res[i, 1] <- repp
      res[i, 2] <- s
      res[i, 3] <- t
      res[i, 4] <- model
      res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      res[i, 6] <- paste0(s, "_", t)
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- s
      res_var[i, 3] <- t
      res_var[i, 4] <- model
      res_var[i, 5] <- dat$model_fit$ETA$G$varU
      res_var[i, 6] <- dat$model_fit$varE
      res_var[i, 7] <- dat$model_fit$ETA$G$varU/(dat$model_fit$ETA$G$varU + dat$model_fit$varE)
      res_var[i, 8] <- paste0(s, "_", t)

    }
  }
}

#res_var_long <- res_var %>% select(rep, VarG, VarEps, env) %>% gather(value="Var", key="Source", VarG, VarEps)
res_var <- transform(res_var, env=as.factor(env))

p_var <- ggplot(res_var, aes(x = env, y = H2)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  labs(x = "Environment", y = expression(italic(PVE)), title="") +
  theme_cowplot(font_size = 18)

print(p_var)



res <- transform(res, env=as.factor(env))

p <- ggplot(res, aes(x = env, y = r2)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  labs(x = "Environment", y = expression(italic(R)^2), title="") +
  theme_cowplot(font_size = 18)

print(p)

res %>% group_by(env) %>% summarise(mean_r2=mean(r2), se_r2=(sd(r2)/n_reps)) %>% as.data.frame()

```

###### Phenotype scaled within environment

```{r random within env scaled within env, fig.height=10, fig.width=13}
sex <- c(1, 0)
temp <- c(18,25,28)
model <- "G"
pheno <- "scaled_within_env_pheno"


i <- 0

n_col <- 6
n_row <- n_reps * length(sex) * length(temp) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "sex", "temp", "model", "r2", "env")

res_var <- as.data.frame(matrix(NA, ncol=n_col+2, nrow=n_row))
colnames(res_var) <- c("rep", "sex", "temp", "model", "VarG", "VarEps", "H2", "env")


for(s in sex){
  for(t in temp){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/G_fit/", prefix, "_", pheno, "_random_within_env_", s, "_", t, "_G_fit_", repp, ".rds"))
      i <- i + 1
      
      res[i, 1] <- repp
      res[i, 2] <- s
      res[i, 3] <- t
      res[i, 4] <- model
      res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      res[i, 6] <- paste0(s, "_", t)
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- s
      res_var[i, 3] <- t
      res_var[i, 4] <- model
      res_var[i, 5] <- dat$model_fit$ETA$G$varU
      res_var[i, 6] <- dat$model_fit$varE
      res_var[i, 7] <- dat$model_fit$ETA$G$varU/(dat$model_fit$ETA$G$varU + dat$model_fit$varE)
      res_var[i, 8] <- paste0(s, "_", t)

    }
  }
}

#res_var_long <- res_var %>% select(rep, VarG, VarEps, env) %>% gather(value="Var", key="Source", VarG, VarEps)
res_var <- transform(res_var, env=as.factor(env))

p_var <- ggplot(res_var, aes(x = env, y = H2)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  labs(x = "Environment", y = expression(italic(PVE)), title="") +
  theme_cowplot(font_size = 18)

print(p_var)



res <- transform(res, env=as.factor(env))

p <- ggplot(res, aes(x = env, y = r2)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  labs(x = "Environment", y = expression(italic(R)^2), title="") +
  theme_cowplot(font_size = 18)

print(p)

res %>% group_by(env) %>% summarise(mean_r2=mean(r2), se_r2=(sd(r2)/n_reps)) %>% as.data.frame()

```

The results show that prediction accuracy varies across environments and is generally low. As expected, scaling the phenotypes does
not affect accuracy in this case.

### Random lines scenario

In this scenario, we combine the data across environments and fit the different models.

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r random lines orig, fig.height=10, fig.width=13}
schemes <- "random_lines"
model <- c("G", "E", "GandE", "GxE")
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

###### Phenotype scaled within environment

```{r random lines within env, fig.height=10, fig.width=13}
schemes <- "random_lines"
model <- c("G", "E", "GandE", "GxE")
pheno <- "scaled_within_env_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With the phenotype on the original scale, the results show that G-BLUP performs poorly. 
This is expected because we are making the implicit assumption that genetic effects are equal across environments, which 
is not the case. E-BLUP does very well in this scenario because when combining the data across environments, a lot of the 
variance is explained by the environmental effects, which we have good power to estimate. GE-BLUP and GxE-BLUP give very 
similar accuracy to E-BLUP. This is not surprising since some lines are not in the training set, so the training-test 
transfer of information happens mostly at the environment level. 

However, with the phenotype scaled within environment, E-BLUP now does not predict at all because we have zeroed out 
differences between environments. All the models including G do pretty poorly. 


##### Heterogeneous residual variance

###### Original scale phenotype

```{r random lines orig het, fig.height=10, fig.width=13}
schemes <- "random_lines"
model <- c('G_het_var','E_het_var','GandE_het_var','GxE_het_var','mvgblup','rrm')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      if(met == "mvgblup"){
        y_test <- do.call("c", dat$y_test)
        yhat_test <- do.call("c", dat$yhat_test)
        res[i, 4] <- summary(lm(y_test ~ yhat_test))$r.squared
      } else {
        res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE GWAS-informed models

##### Homogeneous residual variance

###### Original scale phenotype

```{r random lines orig gwas, fig.height=10, fig.width=13}
schemes <- "random_lines"
model <- 'GxE_gwas'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

... 


##### Heterogeneous residual variance

###### Original scale phenotype

```{r random lines orig het gwas, fig.height=10, fig.width=13}
schemes <- "random_lines"
model <- 'GxE_gwas_het_var'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE by chromosomes models

##### Homogeneous residual variance

###### Original scale phenotype

```{r random lines orig chr, fig.height=10, fig.width=13}
schemes <- "random_lines"
model <- c("GxE", "GxE_by_chr")
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarG_1xE", "VarG_2xE", "VarG_3xE", 
                       "VarG_4xE", "VarG_5xE", "VarG_6xE", "VarGxE", "VarEps", "VarTot")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] <- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] <- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] <- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] <- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] <- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] <- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] <- dat$model_fit$varE
      res_var[i, 14] <- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propG_1xE <- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE <- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE <- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE <- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE <- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE <- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propG_1xE", "propG_2xE", "propG_3xE", 
                                                "propG_4xE", "propG_5xE", "propG_6xE", "propGxE", "propEps"), 
                               labels=c("VarG", "VarE", "VarG_2LxE", "VarG_2RxE", "VarG_3LxE", 
                                        "VarG_3RxE", "VarG_XxE", "VarG_4xE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  #scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

##### Heterogeneous residual variance

###### Original scale phenotype

```{r random lines orig het chr, fig.height=10, fig.width=13}
schemes <- "random_lines"
model <- c('GxE_het_var', 'GxE_by_chr_het_var')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c('GxE_het_var', 'GxE_by_chr_het_var')))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```



### Random observations scenario

In this scenario, every line and environment is potentially represented in the training set. It is specific combinations of
lines and environments (e.g., line_101 in females at 18C) that are not seen in the training set. This is the scenario 
where we expect an advantage from using models with both genotype and environment information because the training-test 
transfer of information happens at every level.

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r random obs orig, fig.height=10, fig.width=13}
schemes <- "random_obs"
model <- c("G", "E", "GandE", "GxE")
pheno <- "original_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

###### Phenotype scaled within environment

```{r random obs within env, fig.height=10, fig.width=13}
schemes <- "random_obs"
model <- c("G", "E", "GandE", "GxE")
pheno <- "scaled_within_env_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With the phenotype on the original scale, the results show that, as expected, GE-BLUP improves 
prediction accuracy over E-BLUP. GxE-BLUP achieves the highest prediction accuracy.

However, with the phenotype scaled within environment, E-BLUP now does not predict at all because we have zeroed out 
differences between environments. G-BLUP and GE-BLUP perform equally well (why does G-BLUP work decently now?) and
GxE-BLUP achieves the highest prediction accuracy.


##### Heterogeneous residual variance

###### Original scale phenotype

```{r random obs orig het, fig.height=10, fig.width=13}
schemes <- "random_obs"
model <- c('G_het_var','E_het_var','GandE_het_var','GxE_het_var','mvgblup','rrm')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      if(met == "mvgblup"){
        y_test <- do.call("c", dat$y_test)
        yhat_test <- do.call("c", dat$yhat_test)
        res[i, 4] <- summary(lm(y_test ~ yhat_test))$r.squared
      } else {
        res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE GWAS-informed models

##### Homogeneous residual variance

###### Original scale phenotype

```{r random obs orig gwas, fig.height=10, fig.width=13}
schemes <- "random_obs"
model <- 'GxE_gwas'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

... 


##### Heterogeneous residual variance

###### Original scale phenotype

```{r random obs orig het gwas, fig.height=10, fig.width=13}
schemes <- "random_obs"
model <- 'GxE_gwas_het_var'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE by chromosomes models

##### Homogeneous residual variance

###### Original scale phenotype

```{r random obs orig chr, fig.height=10, fig.width=13}
schemes <- "random_obs"
model <- c("GxE", "GxE_by_chr")
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarG_1xE", "VarG_2xE", "VarG_3xE", 
                       "VarG_4xE", "VarG_5xE", "VarG_6xE", "VarGxE", "VarEps", "VarTot")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] <- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] <- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] <- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] <- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] <- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] <- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] <- dat$model_fit$varE
      res_var[i, 14] <- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propG_1xE <- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE <- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE <- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE <- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE <- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE <- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propG_1xE", "propG_2xE", "propG_3xE", 
                                                "propG_4xE", "propG_5xE", "propG_6xE", "propGxE", "propEps"), 
                               labels=c("VarG", "VarE", "VarG_2LxE", "VarG_2RxE", "VarG_3LxE", 
                                        "VarG_3RxE", "VarG_XxE", "VarG_4xE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  #scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

##### Heterogeneous residual variance

###### Original scale phenotype

```{r random obs orig het chr, fig.height=10, fig.width=13}
schemes <- "random_obs"
model <- c('GxE_het_var', 'GxE_by_chr_het_var')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c('GxE_het_var', 'GxE_by_chr_het_var')))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```




### New temperature scenario

In this scenario, we assigned all the observations in a specific temperature to the test set, i.e., that temperature 
is never seen in the training set. Thus, there is no transfer of training-test information at the temperature level,
which is the factor that explains the majority of the phenotypic variance.

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across temp orig, fig.height=10, fig.width=13}
n_reps <- 3
schemes <- "temp"
model <- c("G", "E", "GandE", "GxE")
pheno <- "original_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("temp", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("temp", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(temp, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

###### Phenotype scaled within environment

```{r across within env, fig.height=10, fig.width=13}
n_reps <- 3
schemes <- "temp"
model <- c("G", "E", "GandE", "GxE")
pheno <- "scaled_within_env_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("temp", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("temp", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(temp, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

With the phenotype on the original scale, in general, prediction accuracy is lower in this scenario. 
As expected, E-BLUP predicts poorly. G-BLUP predicts decently because all the lines are present in the test set. GE-BLUP improves 
accuracy over G-BLUP. GxE-BLUP is worse than G-BLUP and GE-BLUP, on average. However, when looking at the temperature-specific accuracies, 
we can see that it is only the prediction of lifespan at 28C that has lower accuracy.

With the phenotype scaled within environment, the pattern is very similar in this scenario.


##### Heterogeneous residual variance

###### Original scale phenotype

```{r across temp orig het, fig.height=10, fig.width=13}
n_reps <- 3
schemes <- "temp"
model <- c('G_het_var','E_het_var','GandE_het_var','GxE_het_var','rrm')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE GWAS-informed models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across temp orig gwas, fig.height=10, fig.width=13}
n_reps <- 3
schemes <- "temp"
model <- 'GxE_gwas'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

... 


##### Heterogeneous residual variance

###### Original scale phenotype

```{r across temp orig het gwas, fig.height=10, fig.width=13}
n_reps <- 3
schemes <- "temp"
model <- 'GxE_gwas_het_var'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE by chromosomes models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across temp orig chr, fig.height=10, fig.width=13}
schemes <- "temp"
model <- c("GxE", "GxE_by_chr")
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarG_1xE", "VarG_2xE", "VarG_3xE", 
                       "VarG_4xE", "VarG_5xE", "VarG_6xE", "VarGxE", "VarEps", "VarTot")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] <- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] <- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] <- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] <- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] <- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] <- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] <- dat$model_fit$varE
      res_var[i, 14] <- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propG_1xE <- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE <- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE <- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE <- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE <- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE <- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propG_1xE", "propG_2xE", "propG_3xE", 
                                                "propG_4xE", "propG_5xE", "propG_6xE", "propGxE", "propEps"), 
                               labels=c("VarG", "VarE", "VarG_2LxE", "VarG_2RxE", "VarG_3LxE", 
                                        "VarG_3RxE", "VarG_XxE", "VarG_4xE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  #scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

##### Heterogeneous residual variance

###### Original scale phenotype

```{r across temp orig het chr, fig.height=10, fig.width=13}
schemes <- "temp"
model <- c('GxE_het_var', 'GxE_by_chr_het_var')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c('GxE_het_var', 'GxE_by_chr_het_var')))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```




### New sex scenario

In this scenario, we assigned all the observations in a specific sex to the test set, i.e., that sex 
is never seen in the training set. Thus, there is no transfer of training-test information at the sex level.

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across sex orig, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "sex"
model <- c("G", "E", "GandE", "GxE")
pheno <- "original_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("sex", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("sex", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(sex, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

###### Phenotype scaled within environment

```{r across sex env, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "sex"
model <- c("G", "E", "GandE", "GxE")
pheno <- "scaled_within_env_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("sex", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("sex", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(sex, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

With the phenotype on the original scale, ... 

With the phenotype scaled within environment, ...


##### Heterogeneous residual variance

###### Original scale phenotype

```{r across sex orig het, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "sex"
model <- c('G_het_var','E_het_var','GandE_het_var','GxE_het_var','rrm')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE GWAS-informed models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across sex orig gwas, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "sex"
model <- 'GxE_gwas'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

... 


##### Heterogeneous residual variance

###### Original scale phenotype

```{r across sex orig het gwas, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "sex"
model <- 'GxE_gwas_het_var'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE by chromosomes models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across sex orig chr, fig.height=10, fig.width=13}
schemes <- "sex"
model <- c("GxE", "GxE_by_chr")
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarG_1xE", "VarG_2xE", "VarG_3xE", 
                       "VarG_4xE", "VarG_5xE", "VarG_6xE", "VarGxE", "VarEps", "VarTot")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] <- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] <- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] <- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] <- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] <- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] <- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] <- dat$model_fit$varE
      res_var[i, 14] <- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propG_1xE <- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE <- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE <- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE <- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE <- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE <- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propG_1xE", "propG_2xE", "propG_3xE", 
                                                "propG_4xE", "propG_5xE", "propG_6xE", "propGxE", "propEps"), 
                               labels=c("VarG", "VarE", "VarG_2LxE", "VarG_2RxE", "VarG_3LxE", 
                                        "VarG_3RxE", "VarG_XxE", "VarG_4xE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  #scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

##### Heterogeneous residual variance

###### Original scale phenotype

```{r across sex orig het chr, fig.height=10, fig.width=13}
schemes <- "sex"
model <- c('GxE_het_var', 'GxE_by_chr_het_var')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c('GxE_het_var', 'GxE_by_chr_het_var')))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```




### New temperature and sex scenario

In this scenario, we assigned all the observations in a specific temperature and sex to the test set, i.e., that temperature/sex 
combination is never seen in the training set.

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across temp/sex orig, fig.height=10, fig.width=13}
n_reps <- 6
schemes <- "temp_sex"
model <- c("G", "E", "GandE", "GxE")
pheno <- "original_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("temp", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("temp", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(temp, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

###### Phenotype scaled within environment

```{r across temp/sex within env, fig.height=10, fig.width=13}
n_reps <- 6
schemes <- "temp_sex"
model <- c("G", "E", "GandE", "GxE")
pheno <- "scaled_within_env_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("temp", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("temp", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(temp, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

With the phenotype on the original scale, in general, prediction accuracy is lower in this scenario. 
As expected, E-BLUP does not have any predictive ability; all the observations in the test set get the same predicted phenotype because
they are equally "related" based on the environment. G-BLUP predicts decently because all the lines are present in the test set. GE-BLUP barely 
improves accuracy over G-BLUP. GxE-BLUP is better than G-BLUP and GE-BLUP, on average.

With the phenotype scaled within environment, the pattern is very similar in this scenario, with the improvement of using GxE being larger.


##### Heterogeneous residual variance

###### Original scale phenotype

```{r across temp/sex orig het, fig.height=10, fig.width=13}
n_reps <- 6
schemes <- "temp_sex"
model <- c('G_het_var','E_het_var','GandE_het_var','GxE_het_var','rrm')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE GWAS-informed models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across temp/sex orig gwas, fig.height=10, fig.width=13}
n_reps <- 6
schemes <- "temp_sex"
model <- 'GxE_gwas'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

... 


##### Heterogeneous residual variance

###### Original scale phenotype

```{r across temp/sex orig het gwas, fig.height=10, fig.width=13}
n_reps <- 6
schemes <- "temp_sex"
model <- 'GxE_gwas_het_var'
pheno <- "original_pheno"

i <- 0

n_col <- 5
n_row <- n_reps * length(schemes) * length(pvals)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "pval", "r2")

for(sche in schemes){
  for(met in model){
    for(pv in pvals){
      for(repp in 1:n_reps){
        if(pv=="all"){
          dat <- readRDS(paste0("output/GxE_fit/", prefix, "_", pheno, "_", sche, "_GxE_fit_", repp, ".rds"))
        } else {
          dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met,"_", pv, "_fit_", repp, ".rds"))
        }
        i <- i + 1
        
        res[i, 1] <- repp
        res[i, 2] <- sche
        res[i, 3] <- met
        res[i, 4] <- pv
        if(!is.null(dat)){
          res[i, 5] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
        } else {
          res[i, 5] <- NA
        }
      }
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 pval=factor(pval, levels=pvals))

p <- ggplot(res, aes(x = pval, y = r2, fill = pval)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model, pval) %>% summarise(mean_r2=mean(r2, na.rm = TRUE), se_r=(sd(r2, na.rm = TRUE)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 


#### GxE by chromosomes models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across temp/sex orig chr, fig.height=10, fig.width=13}
schemes <- "temp_sex"
model <- c("GxE", "GxE_by_chr")
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+10, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarG_1xE", "VarG_2xE", "VarG_3xE", 
                       "VarG_4xE", "VarG_5xE", "VarG_6xE", "VarGxE", "VarEps", "VarTot")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_1xE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$G_1xE$varU
      } else {
        res_var[i, 6] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_2xE$varU)){
        res_var[i, 7] <- dat$model_fit$ETA$G_2xE$varU
      } else {
        res_var[i, 7] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_3xE$varU)){
        res_var[i, 8] <- dat$model_fit$ETA$G_3xE$varU
      } else {
        res_var[i, 8] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_4xE$varU)){
        res_var[i, 9] <- dat$model_fit$ETA$G_4xE$varU
      } else {
        res_var[i, 9] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$G_5xE$varU)){
        res_var[i, 10] <- dat$model_fit$ETA$G_5xE$varU
      } else {
        res_var[i, 10] <- NA
      }

      if(!is.null(dat$model_fit$ETA$G_6xE$varU)){
        res_var[i, 11] <- dat$model_fit$ETA$G_6xE$varU
      } else {
        res_var[i, 11] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 12] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 12] <- sum(res_var[i, 6:11], na.rm=TRUE)
      }

      res_var[i, 13] <- dat$model_fit$varE
      res_var[i, 14] <- sum(res_var[i, c(4,5,12,13)], na.rm=TRUE)
    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propG_1xE <- res_var$VarG_1xE/res_var$VarTot
res_var$propG_2xE <- res_var$VarG_2xE/res_var$VarTot
res_var$propG_3xE <- res_var$VarG_3xE/res_var$VarTot
res_var$propG_4xE <- res_var$VarG_4xE/res_var$VarTot
res_var$propG_5xE <- res_var$VarG_5xE/res_var$VarTot
res_var$propG_6xE <- res_var$VarG_6xE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propG_1xE, propG_2xE, propG_3xE, 
                                   propG_4xE, propG_5xE, propG_6xE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propG_1xE", "propG_2xE", "propG_3xE", 
                                                "propG_4xE", "propG_5xE", "propG_6xE", "propGxE", "propEps"), 
                               labels=c("VarG", "VarE", "VarG_2LxE", "VarG_2RxE", "VarG_3LxE", 
                                        "VarG_3RxE", "VarG_XxE", "VarG_4xE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  #scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

##### Heterogeneous residual variance

###### Original scale phenotype

```{r across temp/sex orig het chr, fig.height=10, fig.width=13}
schemes <- "temp_sex"
model <- c('GxE_het_var', 'GxE_by_chr_het_var')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=factor(model, levels = c('GxE_het_var', 'GxE_by_chr_het_var')))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```




### Gradient 1 and 2 scenario

In this scenario, ...

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across grad1/2 orig, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "gradient_1_2"
model <- c("G", "E", "GandE", "GxE")
pheno <- "original_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

###### Phenotype scaled within environment

```{r across grad1/2 within env, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "gradient_1_2"
model <- c("G", "E", "GandE", "GxE")
pheno <- "scaled_within_env_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

With the phenotype on the original scale, ...

With the phenotype scaled within environment, ...


##### Heterogeneous residual variance

###### Original scale phenotype

```{r across grad1/2 orig het, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "gradient_1_2"
model <- c('G_het_var','E_het_var','GandE_het_var','GxE_het_var')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 




### Gradient 3 and 4 scenario

In this scenario, ...

#### Standard models

##### Homogeneous residual variance

###### Original scale phenotype

```{r across grad3/4 orig, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "gradient_3_4"
model <- c("G", "E", "GandE", "GxE")
pheno <- "original_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

###### Phenotype scaled within environment

```{r across grad3/4 within env, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "gradient_3_4"
model <- c("G", "E", "GandE", "GxE")
pheno <- "scaled_within_env_pheno"


i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

res_var <- as.data.frame(matrix(NA, ncol=n_col+4, nrow=n_row))
colnames(res_var) <- c("rep", "scheme", "model", "VarG", "VarE", "VarGxE", "VarEps", "VarTot")


for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
      
      res_var[i, 1] <- repp
      res_var[i, 2] <- sche
      res_var[i, 3] <- met
      if(!is.null(dat$model_fit$ETA$G$varU)){
        res_var[i, 4] <- dat$model_fit$ETA$G$varU
      } else {
        res_var[i, 4] <- NA
      }
      
      if(!is.null(dat$model_fit$ETA$E$varU)){
        res_var[i, 5] <- dat$model_fit$ETA$E$varU
      } else {
        res_var[i, 5] <- NA
      }

      if(!is.null(dat$model_fit$ETA$GxE$varU)){
        res_var[i, 6] <- dat$model_fit$ETA$GxE$varU
      } else {
        res_var[i, 6] <- NA
      }

      res_var[i, 7] <- dat$model_fit$varE
      res_var[i, 8] <- sum(res_var[i, 4:7], na.rm=TRUE)

    }
  }
}


###PVE
res_var$propG <- res_var$VarG/res_var$VarTot
res_var$propE <- res_var$VarE/res_var$VarTot
res_var$propGxE <- res_var$VarGxE/res_var$VarTot
res_var$propEps <- res_var$VarEps/res_var$VarTot

res_var_long <- res_var %>% select(rep, model, propG, propE, propGxE, propEps) %>% 
  gather(value="Prop", key="Source", propG, propE, propGxE, propEps)

res_var_long <- transform(res_var_long, model=factor(model),
                 Source=factor(Source, levels=c("propG", "propE", "propGxE", "propEps"), labels=c("VarG", "VarE", "VarGxE", "VarRes")))

p <- ggplot(res_var_long, aes(x = model, y = Prop, fill = Source)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "total")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(PVE)), fill="Source", title="") +
  theme_cowplot(font_size = 18) 

print(p)

###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()

res %>% group_by(model) %>% as.data.frame()
```

With the phenotype on the original scale, ...

With the phenotype scaled within environment, ...

##### Heterogeneous residual variance

###### Original scale phenotype

```{r across grad3/4 orig het, fig.height=10, fig.width=13}
n_reps <- 2
schemes <- "gradient_3_4"
model <- c('G_het_var','E_het_var','GandE_het_var','GxE_het_var')
pheno <- "original_pheno"

i <- 0

n_col <- 4
n_row <- n_reps * length(schemes) * length(model)
res <- as.data.frame(matrix(NA, ncol=n_col, nrow=n_row))
colnames(res) <- c("rep", "scheme", "model", "r2")

for(sche in schemes){
  for(met in model){
    for(repp in 1:n_reps){
      dat <- readRDS(paste0("output/", met, "_fit/", prefix, "_", pheno, "_", sche, "_", met, "_fit_", repp, ".rds"))
      i <- i + 1
        
      res[i, 1] <- repp
      res[i, 2] <- sche
      res[i, 3] <- met
      res[i, 4] <- summary(lm(dat$y_test ~ dat$yhat_test))$r.squared
    }
  }
}


###Accuracy
res <- transform(res, scheme=as.factor(scheme),
                 model=as.factor(model))

p <- ggplot(res, aes(x = model, y = r2, fill = model)) +
  geom_boxplot(color = "black", outlier.size = 1, width = 0.85) +
  stat_summary(fun=mean, geom="point", shape=23,
               position = position_dodge2(width = 0.87,   
                                          preserve = "single")) +
  scale_fill_manual(values = c("pink", "red", "yellow", "orange", "green", "blue", "lightblue")) +
  labs(x = "Model", y = expression(italic(R)^2), fill="Method", title="") +
  theme_cowplot(font_size = 18) +
  theme(legend.position="none")

print(p)

res %>% group_by(model) %>% summarise(mean_r2=mean(r2), se_r=(sd(r2)/n_reps)) %>% as.data.frame()
```

With heterogeneous residual variance, ... 

